---
title: "Data anlalysis for the project hambiYSKpleiotropy"
author: "Shane Hogle"
date: today
---

## Manuscript:

"TBD"

[Preprint available from bioRxiv]()

## Introduction

This is project contains data from the HAMBI species Long-term Community Selection Experiment (YSK) that Amanda Silvenoinen used in her first PhD thesis experiment. Now Niina Smolander has taken over this project under my supervision.

## The long-term community evolution experiment (YSK)

The YSK experiment has been growing 24 HAMBI species in a community over a long-term period under four treatments:

1.  `bact` is the HAMBI community without any additional environmnetal stress - i.e. the control
2.  `bact_strep` is the HAMBI community with streptomycin (what concentration?)
3.  `bact_pred` is the HAMBI community grown with *Tetrahymena thermophila* ciliate predator
4.  `bact_pred_strep` is the HAMBI community grown with both the predator and the streptomycin.

In YSK 6 biological replicates were established for each of the four conditions above. These 4 $\times$ 6 microcosms have been serially transferred (every 7 days) for now almost 150 transfers (2 years 11 months) allowing species to sort and evolution to occur.

## Experiment overview

After ?? weeks of regular transfers of YSK, Amanda transplanted the 6 $\times$ microcosms per environment into each of the other environments resulting in a full factorial treatment design. She also included a treatment where a naive/ancestral HAMBI community was exposed to YSK conditions. She serially transferred this transplant experiment every 7 days into fresh media (same conditions as YSK) for a total of 12 weeks/transfers (84 days). Considering a weekly dilution factor of 1/10 in the serial transfer series, then $\mathrm{log_2(10)} \approx 3.3$ generations will have elapsed resulting in an evolutionary scale of $3.3 \times 12 \approx 40$ elapsed generations during the transplant experiment.

## Measurements and data types

- Community DNA was sampled at all transfers.
- 16S rRNA amplicon sequencing was performed at transfers 0, 4 (28 days), 8 (56 days), and 12 (84 days)
- Community genome (metagenome) sequencing was performed at transfers **XYZ**
- Ciliate cell density and bacterial optical density ($\mathrm{OD_{600}}$) were measured at every weekly transfer
- After **X** transfers, 32 clones were isolated from each replicate of the transplant microcosms. High-resolution growth curves were obtained for the clones (obtained using the log-phase 600) 
  
## Code

[![DOI](https://zenodo.org/badge/411568623.svg)](https://zenodo.org/badge/latestdoi/411568623) Data and code here is provided under GPL3. Feel free to use or remix as you see fit.

### Project structure
1.  `/R` contains R scripts
2.  `/data` contains data that has been processed in some way for later use
3.  `/_data_raw` contains unprocessed data scraped from compute cluster
4.  `/figs` contains figures generated from R scripts

## Availability

The rendered project site is available at <https://hambidoubleevohi-slhogl-14191cd40bf1aa9a3beb400287f1505ccad06ec.utugit.fi/>. The website has been produced using [Quarto notebooks](https://quarto.org/).

This GitLab repository (<https://gitlab.utu.fi/slhogl/hambiYSKpleiotropy>) hosts the code and data for this project. The rendered webpage can be fully recreated using the code at <https://gitlab.utu.fi/slhogl/hambiYSKpleiotropy>. 

## Reproducibility

The project uses [`renv`](https://rstudio.github.io/renv/index.html) to create reproducible environment to execute the code in this project. [See here](https://rstudio.github.io/renv/articles/renv.html#collaboration) for a brief overview on collaboration and reproduction of the entire project. To get up and running you can do:

``` r
install.packages("renv")
renv::restore()
```
