---
title: "Data analysis for the project hambiEvoVEnv"
date: today
author:
  - name: Shane L Hogle
    orcid: 0000-0003-2595-3863
    affiliation: 
      - name: University of Turku
        city: Turku
        state: Finland
        url: https://www.utu.fi/en
  - name: Niina Smolander
    affiliation: 
      - name: University of Turku
        city: Turku
        state: Finland
        url: https://www.utu.fi/en
#abstract: > 
#  The abstract text can go here 
#  and also here ...
keywords:
  - Microbiology
  - Ecology
  - Evolution
  - Antimicrobial resistance (AMR)
license: "CC BY"
copyright: 
  holder: Shane L Hogle
  year: 2024
# citation: 
#   container-title: BioRxiv
#   volume: 
#   issue: 
#   doi: 
funding: "The author received no specific funding for this work."
bibliography: references.bib
---

## Manuscript:

◇ Corresponding author

‡ equal contributions

### Published record

**A shared evolutionary history shapes community response to the environment.**\
Smolander N‡, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL‡^◇^. *XYZ* (2024/5) [doi:]()

### Preprint

**A shared evolutionary history shapes community response to the environment.**\
Smolander N‡, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL‡^◇^. *BioRxiv* (2024/5) [doi:]()

## Introduction

Microbes exist in highly diverse communities consisting of hundreds of co-occurring species and strains. This biodiversity exists at multiple hierarchical levels: ranging from species diversity, to functional trait diversity, and to genetic diversity within populations of a species . Genetic variation underlies traits that influence bacterial fitness, providing a mechanistic link between diversity at genetic scale and diversity at ecosystem scale. However, the degree to which fine-scale genetic variation impacts microbial population dynamics and species interactions is incompletely understood. Because of complexities of the genotype-phenotype adaptive landscape the presence of genetic diversity within a population may produce unpredictable ecological outcomes [@fridley2007; @lankau2007]. At the same time, the presence of genetic diversity across different species and the potential for natural selection imposed via species interactions adds further dimensions of complexity [@scheuerl2020a; @good2023; @evans2020]. This raises a fundamental question about if and how a microbial community’s response to rapidly changing conditions will depend upon the intraspecific genetic/trait diversity accumulated over past phases of evolution. The answer will be important for designing robust microbiome therapeutics and other microbial systems with resilient functional properties. This question also holds broad implications for the predictability of ecosystems over evolutionary timescales, especially if we wish to forecast future ecosystem states in response to the pressures of global change [@steffen2005].

There are a considerable number of interacting factors governing how community-wide genomic and trait diversity generated under one selective regime will respond under new selective regimes. First, consider a bacterial community assembled from a clonal species pool and that evolves under non-fluctuating selection imposed by its environment. This will first drive ecological sorting favoring species that are inherently more fit, potentially driving others to extinction. Over time it will select for adaptive traits arising within populations. These evolutionary dynamics are governed by the availability of genetic variation produced by mutation. Some of these mutations may eventually rise to fixation replacing their ancestral form, while others diverge into coexisting lineages [@mcenany2024], for example due to clonal interference [@harris2021; @devisser2006; @amicone2021] and multiple mutations [@desai2007]. Ultimately these mutations can feed back into ecological species sorting process completing the eco-evolutionary loop.

Next, consider this genotypically/phenotypically diverse bacterial community encounters a new selective regime. Like in the first scenario ecological factors will sort species to match their fitness to the environment, but now selection also operates upon existing and de novo mutational variation. Existing genotypes with strongly negative fitness tradeoffs (i.e., antagonistic pleiotropy) under the new selective regime should rapidly decrease [@chen2020], while species with antagonistic pleiotropic mutations that swept to fixation under the prior conditions may be at risk for extinction under the new regime unless they can be rescued by mutations with a compensatory fitness effect [@qian2012]. The fate of de novo mutations in the new environment will also depend upon the genomic background of the parent genotype (i.e., epistasis) [@johnson2023]. If the fitness effects of de novo adaptive mutations are sufficiently independent of the parent genotype in which they emerge, selection would be expected to converge upon a similarly fit genotype/phenotype regardless of the parent genotype. However, the fitness effects of mutations are generally mediated by their genomic background [@quandt2013; @weinreich2006]. Thus, genomic backgrounds formed under prior selective regimes may substantially affect the fitness effects of mutations arising under new conditions. Furthermore, If the metabolic networks of different microbial species strongly interact (e.g., cross feeding) then cross-species epistatic effects may also occur [@hablützel2025; @wade2007].

Following this line of reasoning we expect that two communities with different evolutionary histories will display divergent ecological (i.e., composition) and evolutionary responses to a new selective regime if the relative strength of community-wide epistasis and antagonistic pleiotropy is substantial. If these forces have a small community-wide impact we expect a convergent ecological/evolutionary response regardless of a community’s evolutionary history. To test these expectations experimentally we serially propagated a bacterial community, starting from clones of 23 species, for 406 days in two different selective regimes: a low-carbon, complex growth medium with and without the antibiotic streptomycin which we refer to as “evolved communities.” We then reciprocally transplanted each of these evolved communities (spiked with a small amount of the clonal ancestors of each species to ensure no species went extinct) and the clonal ancestor community into the different selective regimes. We allowed the experimental communities to develop for a further 84 days, measuring community composition at three different times and sequencing community genomes from the end of the experiment, which allowed us to track the fate of mutations from the evolved communities in their new environments (Fig. 1).

## Experiment overview

![Structure of the community evolution and transplantation experiments. Exp = experiment, MetaG = metagenomic.](figs/strep_only/exp_design_v4.png)

We first serially propagated a bacterial community, starting from clones of 23 soil and aquatic species [@hogle2024], for 406 days in two different conditions: a low-carbon, complex growth medium with and without the antibiotic streptomycin. We refer to these as “evolved” communities. We isolated 32 clones from each evovlved microcosm and phenotyped them both home environments to determine mean community fitness in each environment. We then reciprocally transplanted each of these evolved communities into the other’s home environment, with a clonal ancestor community serving as a reference. We allowed the experimental communities to grow for a further 84 days, measuring community composition at three different points and sequencing community genomes from the end of the experiment, which allowed us to track the fate of mutations from the evolved communities in their new environments.

### Measurements and data types

-   32 clones were isolated from each replicate microcosm of the community evolution pre-experiment. High-resolution growth curves were obtained for the clones (obtained using the log-phase 600)
-   Bacterial optical density ($\mathrm{OD_{600}}$) were measured at every weekly transfer
-   Community DNA was sampled at all transfers.
-   16S rRNA amplicon sequencing was performed at transfers 0, 4 (28 days), 8 (56 days), and 12 (84 days)
-   Community genome (metagenome) sequencing was performed at transfer 12 of the transplantation experiment and of the evolution treatments starting materials

## Analysis overview

These notebooks allow the full reproduction of the data analysis from the sequencing data and clone phenotyping data. They include analyses of mean community fitness (from clone phenotypcing, for ploting species composition, for performing ordination on community composition, and machine learning to predict community history from the present community composition.

## Availability

Data and code in this GitHub repository (<https://github.com/slhogle/hambiEvoVEnv>) is provided under [GNU AGPL3](https://www.gnu.org/licenses/agpl-3.0.html). The rendered project site is available at <https://slhogle.github.io/hambiEvoVEnv/>, which has been produced using [Quarto notebooks](https://quarto.org/). The content on the rendered site is released under the [CC BY 4.0.](https://creativecommons.org/licenses/by/4.0/) This repository hosts all code and data for this project including the code necessary to fully recreate the rendered webpage.

An archived release of the code here is available from Zenodo:

Raw sequencing data using in the project is available from NCBI Bioproject [ ]().

## Reproducibility

The project uses [`renv`](https://rstudio.github.io/renv/index.html) to create reproducible environment to execute the code in this project. [See here](https://rstudio.github.io/renv/articles/renv.html#collaboration) for a brief overview on collaboration and reproduction of the entire project. To get up and running you can do:

``` r
install.packages("renv")
renv::restore()
```

```{r}
#| eval: true
#| echo: false
#| output: false
#| error: false
#| warning: false
# Not included in rendered output
# Setup project structure
library(here)
library(fs)

# define directory paths and overall structure
# the _data_raw directory is never touched or modified! All downstream work 
# and intermediate results should be put in the data directory

# note that we have split up the structure by whether or not we are focusing only
# on streptomycin experiments or on streptomycin plus predator

## Only streptomycin analyses
fs::dir_create(here::here("data", "strep_only", "communities", "amplicon_sp_counts"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_only", "communities", "cell_density"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_only", "communities", "machine_learning_pipelines"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_only", "monocultures", "clone_phenotyping"), recurse = TRUE)

fs::dir_create(here::here("R", "strep_only", "communities", "amplicon_sp_counts"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_only", "communities", "cell_density"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_only", "communities", "machine_learning_pipelines"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_only", "monocultures", "clone_phenotyping"), recurse = TRUE)

fs::dir_create(here::here("figs", "strep_only"), recurse = TRUE)

## strep and predator analyses
fs::dir_create(here::here("data", "strep_predator", "communities", "amplicon_sp_counts"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_predator", "communities", "cell_density"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_predator", "communities", "machine_learning_pipelines"), recurse = TRUE)
fs::dir_create(here::here("data", "strep_predator", "monocultures", "clone_phenotyping"), recurse = TRUE)

fs::dir_create(here::here("R", "strep_predator", "communities", "amplicon_sp_counts"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_predator", "communities", "cell_density"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_predator", "communities", "machine_learning_pipelines"), recurse = TRUE)
fs::dir_create(here::here("R", "strep_predator", "monocultures", "clone_phenotyping"), recurse = TRUE)

fs::dir_create(here::here("figs", "strep_predator"), recurse = TRUE)
```
