<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane L Hogle">
<meta name="author" content="Niina Smolander">
<meta name="dcterms.date" content="2024-11-03">
<meta name="keywords" content="Microbiology, Ecology, Evolution, Antimicrobial resistance (AMR)">

<title>hambiYSKpleiotropy - Data analysis for the project hambiEvoVEnv</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">
      hambiYSKpleiotropy
      </li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">hambiYSKpleiotropy</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">1. Community densities workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/communities/cell_density/01_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot densities</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">2. Community composition workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Amplicon data wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/communities/amplicon_sp_counts/02_composition_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Community composition</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/communities/machine_learning_pipelines/01_machine_learn_home_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Machine learning the home environment</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">3. Clones workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/monocultures/clone_phenotyping/01_growth_curve_qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect/Smooth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/monocultures/clone_phenotyping/02_growthrate_auc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Summarize curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_only/monocultures/clone_phenotyping/03_growth_summary_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Analysis and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">1A. Community densities workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/communities/cell_density/01_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot densities</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">2A. Community composition workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/communities/amplicon_sp_counts/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Amplicon data wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/communities/amplicon_sp_counts/02_composition_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Community composition</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/communities/machine_learning_pipelines/01_machine_learn_home_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Machine learning the home environment</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true">
 <span class="menu-text">3A. Clones workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/monocultures/clone_phenotyping/01_growth_curve_qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect/Smooth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/monocultures/clone_phenotyping/02_growthrate_auc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Summarize curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/strep_predator/monocultures/clone_phenotyping/03_growth_summary_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Analysis and plot</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#manuscript" id="toc-manuscript" class="nav-link active" data-scroll-target="#manuscript"><span class="header-section-number">1</span> Manuscript:</a>
  <ul class="collapse">
  <li><a href="#published-record" id="toc-published-record" class="nav-link" data-scroll-target="#published-record"><span class="header-section-number">1.1</span> Published record</a></li>
  <li><a href="#preprint" id="toc-preprint" class="nav-link" data-scroll-target="#preprint"><span class="header-section-number">1.2</span> Preprint</a></li>
  </ul></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a></li>
  <li><a href="#experiment-overview" id="toc-experiment-overview" class="nav-link" data-scroll-target="#experiment-overview"><span class="header-section-number">3</span> Experiment overview</a>
  <ul class="collapse">
  <li><a href="#measurements-and-data-types" id="toc-measurements-and-data-types" class="nav-link" data-scroll-target="#measurements-and-data-types"><span class="header-section-number">3.1</span> Measurements and data types</a></li>
  </ul></li>
  <li><a href="#analysis-overview" id="toc-analysis-overview" class="nav-link" data-scroll-target="#analysis-overview"><span class="header-section-number">4</span> Analysis overview</a></li>
  <li><a href="#availability" id="toc-availability" class="nav-link" data-scroll-target="#availability"><span class="header-section-number">5</span> Availability</a></li>
  <li><a href="#reproducibility" id="toc-reproducibility" class="nav-link" data-scroll-target="#reproducibility"><span class="header-section-number">6</span> Reproducibility</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Data analysis for the project hambiEvoVEnv</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Shane L Hogle <a href="https://orcid.org/0000-0003-2595-3863" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://www.utu.fi/en">
            University of Turku
            </a>
          </p>
      </div>
    <div class="quarto-title-meta-contents">
    <p class="author">Niina Smolander </p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://www.utu.fi/en">
            University of Turku
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  

<div>
  <div class="keywords">
    <div class="block-title">Keywords</div>
    <p>Microbiology, Ecology, Evolution, Antimicrobial resistance (AMR)</p>
  </div>
</div>

</header>


<section id="manuscript" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="manuscript"><span class="header-section-number">1</span> Manuscript:</h2>
<p>◇ Corresponding author</p>
<section id="published-record" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="published-record"><span class="header-section-number">1.1</span> Published record</h3>
<p><strong>A shared history shapes community response to the environment.</strong><br>
Smolander N, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL<sup>◇</sup>. <em>XYZ</em> (2024/5) <a href="">doi:</a></p>
</section>
<section id="preprint" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="preprint"><span class="header-section-number">1.2</span> Preprint</h3>
<p><strong>A shared history shapes community response to the environment.</strong><br>
Smolander N, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL<sup>◇</sup>. <em>BioRxiv</em> (2024/5) <a href="">doi:</a></p>
</section>
</section>
<section id="introduction" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="introduction"><span class="header-section-number">2</span> Introduction</h2>
<p>We consider the material and information legacies embedded within a synthetic microbial community that has been serially propagated for over 400 days. We ask whether ecological and evolutionary patterns exhibited by a microbial community in its long-term home environment are retained in a novel environment. Recent work has shown that when entire microbial communities are mixed or transplanted, they retain some degree of cohesion in function and composition <span class="citation" data-cites="rillig2015">(<a href="#ref-rillig2015" role="doc-biblioref">Rillig et al. 2015</a>)</span>. What causes communities to behave cohesively rather than collapsing into their constituents when subjected to abiotic or biotic pressures? When might we expect a cohesive community response instead of the aggregate responses of individual species?</p>
</section>
<section id="experiment-overview" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="experiment-overview"><span class="header-section-number">3</span> Experiment overview</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figs/strep_only/exp_design.png" class="img-fluid figure-img"></p>
<figcaption>Structure of the community evolution and transplantation experiments. Exp = experiment, MetaG = metagenomic.</figcaption>
</figure>
</div>
<p>We first serially propagated a bacterial community, starting from clones of 23 soil and aquatic species <span class="citation" data-cites="hogle2024">(<a href="#ref-hogle2024" role="doc-biblioref">Hogle, Tamminen, and Hiltunen 2024</a>)</span>, for 406 days in two different conditions: a low-carbon, complex growth medium with and without the antibiotic streptomycin. We refer to these as “evolved” communities. We isolated 32 clones from each evovlved microcosm and phenotyped them both home environments to determine mean community fitness in each environment. We then reciprocally transplanted each of these evolved communities into the other’s home environment, with a clonal ancestor community serving as a reference. We allowed the experimental communities to grow for a further 84 days, measuring community composition at three different points and sequencing community genomes from the end of the experiment, which allowed us to track the fate of mutations from the evolved communities in their new environments.</p>
<section id="measurements-and-data-types" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="measurements-and-data-types"><span class="header-section-number">3.1</span> Measurements and data types</h3>
<ul>
<li>32 clones were isolated from each replicate microcosm of the community evolution pre-experiment. High-resolution growth curves were obtained for the clones (obtained using the log-phase 600)</li>
<li>Bacterial optical density (<span class="math inline">\(\mathrm{OD_{600}}\)</span>) were measured at every weekly transfer</li>
<li>Community DNA was sampled at all transfers.</li>
<li>16S rRNA amplicon sequencing was performed at transfers 0, 4 (28 days), 8 (56 days), and 12 (84 days)</li>
<li>Community genome (metagenome) sequencing was performed at transfer 12 of the transplantation experiment and of the evolution treatments starting materials</li>
</ul>
</section>
</section>
<section id="analysis-overview" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="analysis-overview"><span class="header-section-number">4</span> Analysis overview</h2>
<p>These notebooks allow the full reproduction of the data analysis from the sequencing data and clone phenotyping data. They include analyses of mean community fitness (from clone phenotypcing, for ploting species composition, for performing ordination on community composition, and machine learning to predict community history from the present community composition.</p>
</section>
<section id="availability" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="availability"><span class="header-section-number">5</span> Availability</h2>
<p>Data and code in this GitHub repository (<a href="https://github.com/slhogle/hambiEvoVEnv" class="uri">https://github.com/slhogle/hambiEvoVEnv</a>) is provided under <a href="https://www.gnu.org/licenses/agpl-3.0.html">GNU AGPL3</a>. The rendered project site is available at <a href="https://slhogle.github.io/hambiEvoVEnv/" class="uri">https://slhogle.github.io/hambiEvoVEnv/</a>, which has been produced using <a href="https://quarto.org/">Quarto notebooks</a>. The content on the rendered site is released under the <a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0.</a> This repository hosts all code and data for this project including the code necessary to fully recreate the rendered webpage.</p>
<p>An archived release of the code here is available from Zenodo:</p>
<p>Raw sequencing data using in the project is available from NCBI Bioproject <a href=""></a>.</p>
</section>
<section id="reproducibility" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="reproducibility"><span class="header-section-number">6</span> Reproducibility</h2>
<p>The project uses <a href="https://rstudio.github.io/renv/index.html"><code>renv</code></a> to create reproducible environment to execute the code in this project. <a href="https://rstudio.github.io/renv/articles/renv.html#collaboration">See here</a> for a brief overview on collaboration and reproduction of the entire project. To get up and running you can do:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-hogle2024" class="csl-entry" role="listitem">
Hogle, Shane L., Manu Tamminen, and Teppo Hiltunen. 2024. <span>“Complete Genome Sequences of 30 Bacterial Species from a Synthetic Community.”</span> Edited by Elinne Becket. <em>Microbiology Resource Announcements</em> 13 (6). <a href="https://doi.org/10.1128/mra.00111-24">https://doi.org/10.1128/mra.00111-24</a>.
</div>
<div id="ref-rillig2015" class="csl-entry" role="listitem">
Rillig, Matthias C., Janis Antonovics, Tancredi Caruso, Anika Lehmann, Jeff R. Powell, Stavros D. Veresoglou, and Erik Verbruggen. 2015. <span>“Interchange of Entire Communities: Microbial Community Coalescence.”</span> <em>Trends in Ecology &amp; Evolution</em> 30 (8): 470–76. <a href="https://doi.org/10.1016/j.tree.2015.06.004">https://doi.org/10.1016/j.tree.2015.06.004</a>.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a></div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb2" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">---</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="an">title:</span><span class="co"> "Data analysis for the project hambiEvoVEnv"</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="an">author:</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co">  - name: Shane L Hogle</span></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co">    orcid: 0000-0003-2595-3863</span></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="co">    affiliation: </span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co">      - name: University of Turku</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="co">        city: Turku</span></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="co">        state: Finland</span></span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="co">        url: https://www.utu.fi/en</span></span>
<span id="cb2-12"><a href="#cb2-12"></a><span class="co">  - name: Niina Smolander</span></span>
<span id="cb2-13"><a href="#cb2-13"></a><span class="co">    affiliation: </span></span>
<span id="cb2-14"><a href="#cb2-14"></a><span class="co">      - name: University of Turku</span></span>
<span id="cb2-15"><a href="#cb2-15"></a><span class="co">        city: Turku</span></span>
<span id="cb2-16"><a href="#cb2-16"></a><span class="co">        state: Finland</span></span>
<span id="cb2-17"><a href="#cb2-17"></a><span class="co">        url: https://www.utu.fi/en</span></span>
<span id="cb2-18"><a href="#cb2-18"></a><span class="co">#abstract: &gt; </span></span>
<span id="cb2-19"><a href="#cb2-19"></a><span class="co">#  The abstract text can go here </span></span>
<span id="cb2-20"><a href="#cb2-20"></a><span class="co">#  and also here ...</span></span>
<span id="cb2-21"><a href="#cb2-21"></a><span class="an">keywords:</span></span>
<span id="cb2-22"><a href="#cb2-22"></a><span class="co">  - Microbiology</span></span>
<span id="cb2-23"><a href="#cb2-23"></a><span class="co">  - Ecology</span></span>
<span id="cb2-24"><a href="#cb2-24"></a><span class="co">  - Evolution</span></span>
<span id="cb2-25"><a href="#cb2-25"></a><span class="co">  - Antimicrobial resistance (AMR)</span></span>
<span id="cb2-26"><a href="#cb2-26"></a><span class="an">license:</span><span class="co"> "CC BY"</span></span>
<span id="cb2-27"><a href="#cb2-27"></a><span class="an">copyright:</span><span class="co"> </span></span>
<span id="cb2-28"><a href="#cb2-28"></a><span class="co">  holder: Shane L Hogle</span></span>
<span id="cb2-29"><a href="#cb2-29"></a><span class="co">  year: 2024</span></span>
<span id="cb2-30"><a href="#cb2-30"></a><span class="co"># citation: </span></span>
<span id="cb2-31"><a href="#cb2-31"></a><span class="co">#   container-title: BioRxiv</span></span>
<span id="cb2-32"><a href="#cb2-32"></a><span class="co">#   volume: </span></span>
<span id="cb2-33"><a href="#cb2-33"></a><span class="co">#   issue: </span></span>
<span id="cb2-34"><a href="#cb2-34"></a><span class="co">#   doi: </span></span>
<span id="cb2-35"><a href="#cb2-35"></a><span class="an">funding:</span><span class="co"> "The author received no specific funding for this work."</span></span>
<span id="cb2-36"><a href="#cb2-36"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb2-37"><a href="#cb2-37"></a><span class="co">---</span></span>
<span id="cb2-38"><a href="#cb2-38"></a></span>
<span id="cb2-39"><a href="#cb2-39"></a><span class="fu">## Manuscript:</span></span>
<span id="cb2-40"><a href="#cb2-40"></a></span>
<span id="cb2-41"><a href="#cb2-41"></a>◇ Corresponding author</span>
<span id="cb2-42"><a href="#cb2-42"></a></span>
<span id="cb2-43"><a href="#cb2-43"></a><span class="fu">### Published record</span></span>
<span id="cb2-44"><a href="#cb2-44"></a></span>
<span id="cb2-45"><a href="#cb2-45"></a>**A shared history shapes community response to the environment.**\</span>
<span id="cb2-46"><a href="#cb2-46"></a>Smolander N, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL^◇^. *XYZ* (2024/5) <span class="co">[</span><span class="ot">doi:</span><span class="co">]()</span></span>
<span id="cb2-47"><a href="#cb2-47"></a></span>
<span id="cb2-48"><a href="#cb2-48"></a><span class="fu">### Preprint</span></span>
<span id="cb2-49"><a href="#cb2-49"></a></span>
<span id="cb2-50"><a href="#cb2-50"></a>**A shared history shapes community response to the environment.**\</span>
<span id="cb2-51"><a href="#cb2-51"></a>Smolander N, Silvennoinen A, Tamminen M, Hiltunen T, Hogle SL^◇^. *BioRxiv* (2024/5) <span class="co">[</span><span class="ot">doi:</span><span class="co">]()</span></span>
<span id="cb2-52"><a href="#cb2-52"></a></span>
<span id="cb2-53"><a href="#cb2-53"></a><span class="fu">## Introduction</span></span>
<span id="cb2-54"><a href="#cb2-54"></a></span>
<span id="cb2-55"><a href="#cb2-55"></a>We consider the material and information legacies embedded within a synthetic microbial community that has been serially propagated for over 400 days. We ask whether ecological and evolutionary patterns exhibited by a microbial community in its long-term home environment are retained in a novel environment. Recent work has shown that when entire microbial communities are mixed or transplanted, they retain some degree of cohesion in function and composition <span class="co">[</span><span class="ot">@rillig2015</span><span class="co">]</span>. What causes communities to behave cohesively rather than collapsing into their constituents when subjected to abiotic or biotic pressures? When might we expect a cohesive community response instead of the aggregate responses of individual species?</span>
<span id="cb2-56"><a href="#cb2-56"></a></span>
<span id="cb2-57"><a href="#cb2-57"></a><span class="fu">## Experiment overview</span></span>
<span id="cb2-58"><a href="#cb2-58"></a></span>
<span id="cb2-59"><a href="#cb2-59"></a><span class="al">![Structure of the community evolution and transplantation experiments. Exp = experiment, MetaG = metagenomic.](figs/strep_only/exp_design.png)</span></span>
<span id="cb2-60"><a href="#cb2-60"></a></span>
<span id="cb2-61"><a href="#cb2-61"></a>We first serially propagated a bacterial community, starting from clones of 23 soil and aquatic species <span class="co">[</span><span class="ot">@hogle2024</span><span class="co">]</span>, for 406 days in two different conditions: a low-carbon, complex growth medium with and without the antibiotic streptomycin. We refer to these as “evolved” communities. We isolated 32 clones from each evovlved microcosm and phenotyped them both home environments to determine mean community fitness in each environment. We then reciprocally transplanted each of these evolved communities into the other’s home environment, with a clonal ancestor community serving as a reference. We allowed the experimental communities to grow for a further 84 days, measuring community composition at three different points and sequencing community genomes from the end of the experiment, which allowed us to track the fate of mutations from the evolved communities in their new environments.</span>
<span id="cb2-62"><a href="#cb2-62"></a></span>
<span id="cb2-63"><a href="#cb2-63"></a><span class="fu">### Measurements and data types</span></span>
<span id="cb2-64"><a href="#cb2-64"></a></span>
<span id="cb2-65"><a href="#cb2-65"></a><span class="ss">-   </span>32 clones were isolated from each replicate microcosm of the community evolution pre-experiment. High-resolution growth curves were obtained for the clones (obtained using the log-phase 600)</span>
<span id="cb2-66"><a href="#cb2-66"></a><span class="ss">-   </span>Bacterial optical density ($\mathrm{OD_{600}}$) were measured at every weekly transfer</span>
<span id="cb2-67"><a href="#cb2-67"></a><span class="ss">-   </span>Community DNA was sampled at all transfers.</span>
<span id="cb2-68"><a href="#cb2-68"></a><span class="ss">-   </span>16S rRNA amplicon sequencing was performed at transfers 0, 4 (28 days), 8 (56 days), and 12 (84 days)</span>
<span id="cb2-69"><a href="#cb2-69"></a><span class="ss">-   </span>Community genome (metagenome) sequencing was performed at transfer 12 of the transplantation experiment and of the evolution treatments starting materials</span>
<span id="cb2-70"><a href="#cb2-70"></a></span>
<span id="cb2-71"><a href="#cb2-71"></a><span class="fu">## Analysis overview</span></span>
<span id="cb2-72"><a href="#cb2-72"></a></span>
<span id="cb2-73"><a href="#cb2-73"></a>These notebooks allow the full reproduction of the data analysis from the sequencing data and clone phenotyping data. They include analyses of mean community fitness (from clone phenotypcing, for ploting species composition, for performing ordination on community composition, and machine learning to predict community history from the present community composition.</span>
<span id="cb2-74"><a href="#cb2-74"></a></span>
<span id="cb2-75"><a href="#cb2-75"></a><span class="fu">## Availability</span></span>
<span id="cb2-76"><a href="#cb2-76"></a></span>
<span id="cb2-77"><a href="#cb2-77"></a>Data and code in this GitHub repository (<span class="ot">&lt;https://github.com/slhogle/hambiEvoVEnv&gt;</span>) is provided under <span class="co">[</span><span class="ot">GNU AGPL3</span><span class="co">](https://www.gnu.org/licenses/agpl-3.0.html)</span>. The rendered project site is available at <span class="ot">&lt;https://slhogle.github.io/hambiEvoVEnv/&gt;</span>, which has been produced using <span class="co">[</span><span class="ot">Quarto notebooks</span><span class="co">](https://quarto.org/)</span>. The content on the rendered site is released under the <span class="co">[</span><span class="ot">CC BY 4.0.</span><span class="co">](https://creativecommons.org/licenses/by/4.0/)</span> This repository hosts all code and data for this project including the code necessary to fully recreate the rendered webpage.</span>
<span id="cb2-78"><a href="#cb2-78"></a></span>
<span id="cb2-79"><a href="#cb2-79"></a>An archived release of the code here is available from Zenodo:</span>
<span id="cb2-80"><a href="#cb2-80"></a></span>
<span id="cb2-81"><a href="#cb2-81"></a>Raw sequencing data using in the project is available from NCBI Bioproject <span class="co">[ ]()</span>.</span>
<span id="cb2-82"><a href="#cb2-82"></a></span>
<span id="cb2-83"><a href="#cb2-83"></a><span class="fu">## Reproducibility</span></span>
<span id="cb2-84"><a href="#cb2-84"></a></span>
<span id="cb2-85"><a href="#cb2-85"></a>The project uses <span class="co">[</span><span class="ot">`renv`</span><span class="co">](https://rstudio.github.io/renv/index.html)</span> to create reproducible environment to execute the code in this project. <span class="co">[</span><span class="ot">See here</span><span class="co">](https://rstudio.github.io/renv/articles/renv.html#collaboration)</span> for a brief overview on collaboration and reproduction of the entire project. To get up and running you can do:</span>
<span id="cb2-86"><a href="#cb2-86"></a></span>
<span id="cb2-87"><a href="#cb2-87"></a><span class="in">``` r</span></span>
<span id="cb2-88"><a href="#cb2-88"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb2-89"><a href="#cb2-89"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb2-90"><a href="#cb2-90"></a><span class="in">```</span></span>
<span id="cb2-91"><a href="#cb2-91"></a></span>
<span id="cb2-94"><a href="#cb2-94"></a><span class="in">```{r}</span></span>
<span id="cb2-95"><a href="#cb2-95"></a><span class="co">#| eval: true</span></span>
<span id="cb2-96"><a href="#cb2-96"></a><span class="co">#| echo: false</span></span>
<span id="cb2-97"><a href="#cb2-97"></a><span class="co">#| output: false</span></span>
<span id="cb2-98"><a href="#cb2-98"></a><span class="co">#| error: false</span></span>
<span id="cb2-99"><a href="#cb2-99"></a><span class="co">#| warning: false</span></span>
<span id="cb2-100"><a href="#cb2-100"></a><span class="co"># Not included in rendered output</span></span>
<span id="cb2-101"><a href="#cb2-101"></a><span class="co"># Setup project structure</span></span>
<span id="cb2-102"><a href="#cb2-102"></a><span class="fu">library</span>(here)</span>
<span id="cb2-103"><a href="#cb2-103"></a><span class="fu">library</span>(fs)</span>
<span id="cb2-104"><a href="#cb2-104"></a></span>
<span id="cb2-105"><a href="#cb2-105"></a><span class="co"># define directory paths and overall structure</span></span>
<span id="cb2-106"><a href="#cb2-106"></a><span class="co"># the _data_raw directory is never touched or modified! All downstream work </span></span>
<span id="cb2-107"><a href="#cb2-107"></a><span class="co"># and intermediate results should be put in the data directory</span></span>
<span id="cb2-108"><a href="#cb2-108"></a></span>
<span id="cb2-109"><a href="#cb2-109"></a><span class="co"># note that we have split up the structure by whether or not we are focusing only</span></span>
<span id="cb2-110"><a href="#cb2-110"></a><span class="co"># on streptomycin experiments or on streptomycin plus predator</span></span>
<span id="cb2-111"><a href="#cb2-111"></a></span>
<span id="cb2-112"><a href="#cb2-112"></a><span class="do">## Only streptomycin analyses</span></span>
<span id="cb2-113"><a href="#cb2-113"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon_sp_counts"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-114"><a href="#cb2-114"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"cell_density"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-115"><a href="#cb2-115"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"machine_learning_pipelines"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-116"><a href="#cb2-116"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_only"</span>, <span class="st">"monocultures"</span>, <span class="st">"clone_phenotyping"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-117"><a href="#cb2-117"></a></span>
<span id="cb2-118"><a href="#cb2-118"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon_sp_counts"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-119"><a href="#cb2-119"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"cell_density"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-120"><a href="#cb2-120"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_only"</span>, <span class="st">"communities"</span>, <span class="st">"machine_learning_pipelines"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-121"><a href="#cb2-121"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_only"</span>, <span class="st">"monocultures"</span>, <span class="st">"clone_phenotyping"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-122"><a href="#cb2-122"></a></span>
<span id="cb2-123"><a href="#cb2-123"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"strep_only"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-124"><a href="#cb2-124"></a></span>
<span id="cb2-125"><a href="#cb2-125"></a><span class="do">## strep and predator analyses</span></span>
<span id="cb2-126"><a href="#cb2-126"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon_sp_counts"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-127"><a href="#cb2-127"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"cell_density"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-128"><a href="#cb2-128"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"machine_learning_pipelines"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-129"><a href="#cb2-129"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"strep_predator"</span>, <span class="st">"monocultures"</span>, <span class="st">"clone_phenotyping"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-130"><a href="#cb2-130"></a></span>
<span id="cb2-131"><a href="#cb2-131"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon_sp_counts"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-132"><a href="#cb2-132"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"cell_density"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-133"><a href="#cb2-133"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_predator"</span>, <span class="st">"communities"</span>, <span class="st">"machine_learning_pipelines"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-134"><a href="#cb2-134"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"strep_predator"</span>, <span class="st">"monocultures"</span>, <span class="st">"clone_phenotyping"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-135"><a href="#cb2-135"></a></span>
<span id="cb2-136"><a href="#cb2-136"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"strep_predator"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-137"><a href="#cb2-137"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>