<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-03-25">

<title>Data formatting of Rbec output from 20240318_BTK_illumina_v3 – hambiEvoVEnv</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<script src="../../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">Community composition workflow</a></li><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">1) Amplicon data wrangling</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../../">hambiEvoVEnv</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Community densities workflow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/cell_density/01_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1) Process and plot densities</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Community composition workflow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">1) Amplicon data wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/02_composition_plotting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2) Community composition barplot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/03_ordination_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3) Community ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/04_diversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4) Community diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/05_JSD_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5) Community JSD divergences</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Clones workflow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/01_growth_curve_qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1) Inspect/Smooth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/02_growthrate_auc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2) Summarize curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/03_growth_summary_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3) Analysis and plot</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#s-copy-number-information" id="toc-s-copy-number-information" class="nav-link" data-scroll-target="#s-copy-number-information"><span class="header-section-number">1.1</span> 16S copy number information</a></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions"><span class="header-section-number">1.2</span> Functions</a></li>
  </ul></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">2</span> Read data</a>
  <ul class="collapse">
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata"><span class="header-section-number">2.1</span> Metadata</a></li>
  <li><a href="#rbec" id="toc-rbec" class="nav-link" data-scroll-target="#rbec"><span class="header-section-number">2.2</span> Rbec</a></li>
  </ul></li>
  <li><a href="#format" id="toc-format" class="nav-link" data-scroll-target="#format"><span class="header-section-number">3</span> Format</a></li>
  <li><a href="#write-results" id="toc-write-results" class="nav-link" data-scroll-target="#write-results"><span class="header-section-number">4</span> Write results</a></li>
  <li><a href="#clean-up" id="toc-clean-up" class="nav-link" data-scroll-target="#clean-up"><span class="header-section-number">5</span> Clean up</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">Community composition workflow</a></li><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">1) Amplicon data wrangling</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Data formatting of Rbec output from 20240318_BTK_illumina_v3</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
<p class="subtitle lead">Community composition workflow</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">March 25, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook reads in the Rbec output and processes it into species counts tables that are usable in downstream steps. This sequencing run contains both results from hambiYSKpleiotropy experiment and also a test of the boilprep extraction method from the tipping point experiment. The tipping points experimental data is written but not used in this project.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Loads required libraries</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(archive)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Sets some important variables for the analysis. In particular it defines a temporary location on disk to decompress the Rbec tarball</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240318_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="s-copy-number-information" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="s-copy-number-information"><span class="header-section-number">1.1</span> 16S copy number information</h2>
<p>This tibble contains information mapping rRNA 16S locus tags to species and also total copy number of 16S rRNAs in each genome</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>tax_locus_copynum <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb3-2"><a href="#cb3-2"></a>     <span class="sc">~</span>strainID, <span class="sc">~</span>rRNA16S_cn, <span class="sc">~</span>rRNA16S_locus,             <span class="sc">~</span>genus,        <span class="sc">~</span>species,</span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="st">"HAMBI_0006"</span>,          <span class="dv">7</span>L,  <span class="st">"H0006_04757"</span>,      <span class="st">"Pseudomonas"</span>,        <span class="st">"putida"</span>,</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_00044"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_02759"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-6"><a href="#cb3-6"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_01762"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-7"><a href="#cb3-7"></a>  <span class="st">"HAMBI_0105"</span>,          <span class="dv">4</span>L,  <span class="st">"H0105_02306"</span>,    <span class="st">"Agrobacterium"</span>,   <span class="st">"tumefaciens"</span>,</span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="st">"HAMBI_0262"</span>,          <span class="dv">3</span>L,  <span class="st">"H0262_00030"</span>,    <span class="st">"Brevundimonas"</span>,       <span class="st">"bullata"</span>,</span>
<span id="cb3-9"><a href="#cb3-9"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00517"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb3-10"><a href="#cb3-10"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00522"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb3-11"><a href="#cb3-11"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03627"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-12"><a href="#cb3-12"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_00125"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-13"><a href="#cb3-13"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03957"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-14"><a href="#cb3-14"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03997"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03402"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb3-16"><a href="#cb3-16"></a>  <span class="st">"HAMBI_1292"</span>,          <span class="dv">7</span>L,  <span class="st">"H1292_03239"</span>,       <span class="st">"Morganella"</span>,      <span class="st">"morganii"</span>,</span>
<span id="cb3-17"><a href="#cb3-17"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_04293"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-18"><a href="#cb3-18"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_01283"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-19"><a href="#cb3-19"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1279_03957"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-20"><a href="#cb3-20"></a>  <span class="st">"HAMBI_1842"</span>,          <span class="dv">4</span>L,  <span class="st">"H1842_01650"</span>,      <span class="st">"Sphingobium"</span>,    <span class="st">"yanoikuyae"</span>,</span>
<span id="cb3-21"><a href="#cb3-21"></a>  <span class="st">"HAMBI_1896"</span>,          <span class="dv">4</span>L,  <span class="st">"H1896_00963"</span>, <span class="st">"Sphingobacterium"</span>,  <span class="st">"spiritivorum"</span>,</span>
<span id="cb3-22"><a href="#cb3-22"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_00343"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb3-23"><a href="#cb3-23"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_03531"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb3-24"><a href="#cb3-24"></a>  <span class="st">"HAMBI_1977"</span>,          <span class="dv">5</span>L,  <span class="st">"H1977_00118"</span>,      <span class="st">"Pseudomonas"</span>,  <span class="st">"chlororaphis"</span>,</span>
<span id="cb3-25"><a href="#cb3-25"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05160"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-26"><a href="#cb3-26"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05152"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-27"><a href="#cb3-27"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05165"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-28"><a href="#cb3-28"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_01406"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb3-29"><a href="#cb3-29"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_05851"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb3-30"><a href="#cb3-30"></a>  <span class="st">"HAMBI_2160"</span>,          <span class="dv">3</span>L,  <span class="st">"H2160_00530"</span>,       <span class="st">"Bordetella"</span>,         <span class="st">"avium"</span>,</span>
<span id="cb3-31"><a href="#cb3-31"></a>  <span class="st">"HAMBI_2164"</span>,          <span class="dv">5</span>L,  <span class="st">"H2164_03337"</span>,      <span class="st">"Cupriavidus"</span>,    <span class="st">"oxalaticus"</span>,</span>
<span id="cb3-32"><a href="#cb3-32"></a>  <span class="st">"HAMBI_2443"</span>,          <span class="dv">3</span>L,  <span class="st">"H2443_00128"</span>,       <span class="st">"Paracoccus"</span>, <span class="st">"denitrificans"</span>,</span>
<span id="cb3-33"><a href="#cb3-33"></a>  <span class="st">"HAMBI_2494"</span>,          <span class="dv">4</span>L,  <span class="st">"H2494_03389"</span>, <span class="st">"Paraburkholderia"</span>,   <span class="st">"kururiensis"</span>,</span>
<span id="cb3-34"><a href="#cb3-34"></a>  <span class="st">"HAMBI_2659"</span>,          <span class="dv">4</span>L,  <span class="st">"H2659_00367"</span>, <span class="st">"Stenotrophomonas"</span>,   <span class="st">"maltophilia"</span>,</span>
<span id="cb3-35"><a href="#cb3-35"></a>  <span class="st">"HAMBI_2792"</span>,          <span class="dv">4</span>L,  <span class="st">"H2792_00549"</span>,        <span class="st">"Moraxella"</span>,         <span class="st">"canis"</span>,</span>
<span id="cb3-36"><a href="#cb3-36"></a>  <span class="st">"HAMBI_3031"</span>,          <span class="dv">2</span>L,  <span class="st">"H3031_00830"</span>,         <span class="st">"Niabella"</span>,  <span class="st">"yanshanensis"</span>,</span>
<span id="cb3-37"><a href="#cb3-37"></a>  <span class="st">"HAMBI_3237"</span>,          <span class="dv">6</span>L,  <span class="st">"H3237_00875"</span>,       <span class="st">"Microvirga"</span>,   <span class="st">"lotononidis"</span>,</span>
<span id="cb3-38"><a href="#cb3-38"></a>  <span class="st">"HAMBI_1923"</span>,          <span class="dv">6</span>L,  <span class="st">"H1923_00876"</span>,   <span class="st">"Flavobacterium"</span>,      <span class="st">"odoratum"</span></span>
<span id="cb3-39"><a href="#cb3-39"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="functions" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="functions"><span class="header-section-number">1.2</span> Functions</h2>
<p>These are functions to aid in reading and processing output from Rbec</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># this function </span></span>
<span id="cb4-2"><a href="#cb4-2"></a>normalize_by_copy <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum){</span>
<span id="cb4-3"><a href="#cb4-3"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4"></a>    <span class="co"># join with the copy number data frame. We join by the locus tag so this will add H1279_03957 to HAMBI_1299</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(tlc, <span class="at">by =</span> <span class="fu">join_by</span>(rRNA16S_locus), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>    <span class="co"># get total number of mapping reads per species. This aggregates all the difference ASVs per species</span></span>
<span id="cb4-7"><a href="#cb4-7"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(count), <span class="at">.by =</span> <span class="fu">c</span>(sample, strainID, rRNA16S_cn)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8"></a>    <span class="co"># group by sample</span></span>
<span id="cb4-9"><a href="#cb4-9"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10"></a>    <span class="co"># calculate a corrected count which is simply the count divided by copy num for each species</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>    <span class="co"># dividide by the sum of count divided by copy num for whole sample multiplied by the total</span></span>
<span id="cb4-12"><a href="#cb4-12"></a>    <span class="co"># number of mapped reads per sample</span></span>
<span id="cb4-13"><a href="#cb4-13"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count_correct =</span> <span class="fu">round</span>(<span class="fu">sum</span>(count)<span class="sc">*</span>(count<span class="sc">/</span>rRNA16S_cn)<span class="sc">/</span><span class="fu">sum</span>(count<span class="sc">/</span>rRNA16S_cn))) <span class="sc">%&gt;%</span>  </span>
<span id="cb4-14"><a href="#cb4-14"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="#cb4-15"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count, count_correct)</span>
<span id="cb4-16"><a href="#cb4-16"></a>  }</span>
<span id="cb4-17"><a href="#cb4-17"></a></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co"># this function replaces missing species counts with zero</span></span>
<span id="cb4-19"><a href="#cb4-19"></a>completecombos <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum, <span class="at">countname =</span> count, <span class="at">remove1923 =</span> <span class="cn">TRUE</span>){</span>
<span id="cb4-20"><a href="#cb4-20"></a> </span>
<span id="cb4-21"><a href="#cb4-21"></a>  <span class="co"># get unique strainIDs</span></span>
<span id="cb4-22"><a href="#cb4-22"></a>  strainID <span class="ot">&lt;-</span> <span class="fu">unique</span>(tlc<span class="sc">$</span>strainID)</span>
<span id="cb4-23"><a href="#cb4-23"></a>  <span class="co"># table for assigning genus and species names. Doesn't matter if 1923 is there or not</span></span>
<span id="cb4-24"><a href="#cb4-24"></a>  <span class="co"># because it is filter joined later</span></span>
<span id="cb4-25"><a href="#cb4-25"></a>  tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species))</span>
<span id="cb4-26"><a href="#cb4-26"></a>  <span class="cf">if</span> (remove1923) {</span>
<span id="cb4-27"><a href="#cb4-27"></a>    <span class="co"># get unique strainIDs but exclude 1923 if remove1923 is true</span></span>
<span id="cb4-28"><a href="#cb4-28"></a>    strainID <span class="ot">&lt;-</span> strainID[strainID <span class="sc">!=</span> <span class="st">"HAMBI_1923"</span>]</span>
<span id="cb4-29"><a href="#cb4-29"></a>  }</span>
<span id="cb4-30"><a href="#cb4-30"></a>  </span>
<span id="cb4-31"><a href="#cb4-31"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">strainID =</span> strainID, <span class="at">sample =</span> <span class="st">"dummy"</span>), .data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-32"><a href="#cb4-32"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"dummy"</span>, <span class="dv">1</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb4-33"><a href="#cb4-33"></a>    tidyr<span class="sc">::</span><span class="fu">complete</span>(sample, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-34"><a href="#cb4-34"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample <span class="sc">!=</span> <span class="st">"dummy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-35"><a href="#cb4-35"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(<span class="fu">is.na</span>({{ countname }}), <span class="dv">0</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb4-36"><a href="#cb4-36"></a>    tidyr<span class="sc">::</span><span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">count_correct =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-37"><a href="#cb4-37"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species)), <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-38"><a href="#cb4-38"></a>    dplyr<span class="sc">::</span><span class="fu">relocate</span>(genus, species, <span class="at">.after =</span> strainID)</span>
<span id="cb4-39"><a href="#cb4-39"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read data</h1>
<section id="metadata" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="metadata"><span class="header-section-number">2.1</span> Metadata</h2>
<p>Reads metadata for the samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240318_BTK_illumina_v3_metadata.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="rbec" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="rbec"><span class="header-section-number">2.2</span> Rbec</h2>
<p>Untar Rbec output tarball which contains raw counts tables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb6-2"><a href="#cb6-2"></a>  amplicontar,</span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb6-7"><a href="#cb6-7"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Sets up directory structure for decompressing</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb7-3"><a href="#cb7-3"></a>sampnames <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4"></a>  purrr<span class="sc">::</span><span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Reads all the <code>strain_table.txt</code> files in a loop</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(</span>
<span id="cb8-4"><a href="#cb8-4"></a>  readr<span class="sc">::</span>read_tsv,</span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-8"><a href="#cb8-8"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="format" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Format</h1>
<p>Calls function from above that normalize counts by 16S copy number</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>straintabs_norm <span class="ot">&lt;-</span> <span class="fu">normalize_by_copy</span>(straintabs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Calls function from above that complete all combinations of 23 species</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>straintabs_norm_fmt <span class="ot">&lt;-</span> <span class="fu">completecombos</span>(straintabs_norm, <span class="at">tlc =</span> tax_locus_copynum, <span class="at">countname =</span> count, <span class="at">remove1923 =</span> <span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Joins the count data and metadata</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>finaltable <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(straintabs_norm_fmt, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Parition out the two different projects (tipping points and pleiotropy were sequenced on same flow cell) and filter the pleiotropy samples to only include samples for ancestral community, YSK bacteria, and YSK streptomycin</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>finaltip <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">filter</span>(project <span class="sc">==</span> <span class="st">"tipping_points"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pool<span class="sc">:</span>project), <span class="sc">-</span>lib_prep, <span class="sc">-</span>measure_env, <span class="sc">-</span>evolution_env)</span>
<span id="cb12-4"><a href="#cb12-4"></a></span>
<span id="cb12-5"><a href="#cb12-5"></a>finalpleio <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">filter</span>(project <span class="sc">==</span> <span class="st">"amanda_chapt01"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pool<span class="sc">:</span>project), <span class="sc">-</span>lib_prep, <span class="sc">-</span>evo_hist, <span class="sc">-</span>amp_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">filter</span>(measure_env <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"none"</span>, <span class="st">"bact"</span>, <span class="st">"bact_strep"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9"></a>  <span class="fu">filter</span>(evolution_env <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"bact"</span>, <span class="st">"bact_strep"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="write-results" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Write results</h1>
<p>Write table of species abundances with metadata</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>readr<span class="sc">:::</span><span class="fu">write_tsv</span>(finalpleio, here<span class="sc">::</span><span class="fu">here</span>(data_sp, <span class="st">"species_counts_md.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="clean-up" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Clean up</h1>
<p>Remove decompressed coverage directory from temp location</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb15" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb15-1"><a href="#cb15-1"></a><span class="co">---</span></span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="an">title:</span><span class="co"> "Data formatting of Rbec output from 20240318_BTK_illumina_v3"</span></span>
<span id="cb15-3"><a href="#cb15-3"></a><span class="an">subtitle:</span><span class="co"> "Community composition workflow"</span></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb15-6"><a href="#cb15-6"></a><span class="an">abstract:</span><span class="co"> "This notebook reads in the Rbec output and processes it into species counts tables that are usable in downstream steps. This sequencing run contains both results from hambiYSKpleiotropy experiment and also a test of the boilprep extraction method from the tipping point experiment. The tipping points experimental data is written but not used in this project."</span></span>
<span id="cb15-7"><a href="#cb15-7"></a><span class="co">---</span></span>
<span id="cb15-8"><a href="#cb15-8"></a></span>
<span id="cb15-9"><a href="#cb15-9"></a><span class="fu"># Setup</span></span>
<span id="cb15-10"><a href="#cb15-10"></a></span>
<span id="cb15-11"><a href="#cb15-11"></a>Loads required libraries</span>
<span id="cb15-12"><a href="#cb15-12"></a></span>
<span id="cb15-15"><a href="#cb15-15"></a><span class="in">```{r}</span></span>
<span id="cb15-16"><a href="#cb15-16"></a><span class="co">#| output: false</span></span>
<span id="cb15-17"><a href="#cb15-17"></a><span class="co">#| warning: false</span></span>
<span id="cb15-18"><a href="#cb15-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-19"><a href="#cb15-19"></a><span class="fu">library</span>(here)</span>
<span id="cb15-20"><a href="#cb15-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb15-21"><a href="#cb15-21"></a><span class="fu">library</span>(archive)</span>
<span id="cb15-22"><a href="#cb15-22"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb15-23"><a href="#cb15-23"></a><span class="in">```</span></span>
<span id="cb15-24"><a href="#cb15-24"></a></span>
<span id="cb15-25"><a href="#cb15-25"></a>Sets some important variables for the analysis. In particular it defines a temporary location on disk to decompress the Rbec tarball</span>
<span id="cb15-26"><a href="#cb15-26"></a></span>
<span id="cb15-29"><a href="#cb15-29"></a><span class="in">```{r}</span></span>
<span id="cb15-30"><a href="#cb15-30"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240318_BTK_illumina_v3"</span>)</span>
<span id="cb15-31"><a href="#cb15-31"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb15-32"><a href="#cb15-32"></a></span>
<span id="cb15-33"><a href="#cb15-33"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb15-34"><a href="#cb15-34"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span>
<span id="cb15-35"><a href="#cb15-35"></a><span class="in">```</span></span>
<span id="cb15-36"><a href="#cb15-36"></a></span>
<span id="cb15-37"><a href="#cb15-37"></a><span class="fu">## 16S copy number information</span></span>
<span id="cb15-38"><a href="#cb15-38"></a></span>
<span id="cb15-39"><a href="#cb15-39"></a>This tibble contains information mapping rRNA 16S locus tags to species and also total copy number of 16S rRNAs in each genome</span>
<span id="cb15-40"><a href="#cb15-40"></a></span>
<span id="cb15-43"><a href="#cb15-43"></a><span class="in">```{r}</span></span>
<span id="cb15-44"><a href="#cb15-44"></a>tax_locus_copynum <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb15-45"><a href="#cb15-45"></a>     <span class="sc">~</span>strainID, <span class="sc">~</span>rRNA16S_cn, <span class="sc">~</span>rRNA16S_locus,             <span class="sc">~</span>genus,        <span class="sc">~</span>species,</span>
<span id="cb15-46"><a href="#cb15-46"></a>  <span class="st">"HAMBI_0006"</span>,          <span class="dv">7</span>L,  <span class="st">"H0006_04757"</span>,      <span class="st">"Pseudomonas"</span>,        <span class="st">"putida"</span>,</span>
<span id="cb15-47"><a href="#cb15-47"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_00044"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb15-48"><a href="#cb15-48"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_02759"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb15-49"><a href="#cb15-49"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_01762"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb15-50"><a href="#cb15-50"></a>  <span class="st">"HAMBI_0105"</span>,          <span class="dv">4</span>L,  <span class="st">"H0105_02306"</span>,    <span class="st">"Agrobacterium"</span>,   <span class="st">"tumefaciens"</span>,</span>
<span id="cb15-51"><a href="#cb15-51"></a>  <span class="st">"HAMBI_0262"</span>,          <span class="dv">3</span>L,  <span class="st">"H0262_00030"</span>,    <span class="st">"Brevundimonas"</span>,       <span class="st">"bullata"</span>,</span>
<span id="cb15-52"><a href="#cb15-52"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00517"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb15-53"><a href="#cb15-53"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00522"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb15-54"><a href="#cb15-54"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03627"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb15-55"><a href="#cb15-55"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_00125"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb15-56"><a href="#cb15-56"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03957"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb15-57"><a href="#cb15-57"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03997"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb15-58"><a href="#cb15-58"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03402"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb15-59"><a href="#cb15-59"></a>  <span class="st">"HAMBI_1292"</span>,          <span class="dv">7</span>L,  <span class="st">"H1292_03239"</span>,       <span class="st">"Morganella"</span>,      <span class="st">"morganii"</span>,</span>
<span id="cb15-60"><a href="#cb15-60"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_04293"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb15-61"><a href="#cb15-61"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_01283"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb15-62"><a href="#cb15-62"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1279_03957"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb15-63"><a href="#cb15-63"></a>  <span class="st">"HAMBI_1842"</span>,          <span class="dv">4</span>L,  <span class="st">"H1842_01650"</span>,      <span class="st">"Sphingobium"</span>,    <span class="st">"yanoikuyae"</span>,</span>
<span id="cb15-64"><a href="#cb15-64"></a>  <span class="st">"HAMBI_1896"</span>,          <span class="dv">4</span>L,  <span class="st">"H1896_00963"</span>, <span class="st">"Sphingobacterium"</span>,  <span class="st">"spiritivorum"</span>,</span>
<span id="cb15-65"><a href="#cb15-65"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_00343"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb15-66"><a href="#cb15-66"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_03531"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb15-67"><a href="#cb15-67"></a>  <span class="st">"HAMBI_1977"</span>,          <span class="dv">5</span>L,  <span class="st">"H1977_00118"</span>,      <span class="st">"Pseudomonas"</span>,  <span class="st">"chlororaphis"</span>,</span>
<span id="cb15-68"><a href="#cb15-68"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05160"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb15-69"><a href="#cb15-69"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05152"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb15-70"><a href="#cb15-70"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05165"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb15-71"><a href="#cb15-71"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_01406"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb15-72"><a href="#cb15-72"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_05851"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb15-73"><a href="#cb15-73"></a>  <span class="st">"HAMBI_2160"</span>,          <span class="dv">3</span>L,  <span class="st">"H2160_00530"</span>,       <span class="st">"Bordetella"</span>,         <span class="st">"avium"</span>,</span>
<span id="cb15-74"><a href="#cb15-74"></a>  <span class="st">"HAMBI_2164"</span>,          <span class="dv">5</span>L,  <span class="st">"H2164_03337"</span>,      <span class="st">"Cupriavidus"</span>,    <span class="st">"oxalaticus"</span>,</span>
<span id="cb15-75"><a href="#cb15-75"></a>  <span class="st">"HAMBI_2443"</span>,          <span class="dv">3</span>L,  <span class="st">"H2443_00128"</span>,       <span class="st">"Paracoccus"</span>, <span class="st">"denitrificans"</span>,</span>
<span id="cb15-76"><a href="#cb15-76"></a>  <span class="st">"HAMBI_2494"</span>,          <span class="dv">4</span>L,  <span class="st">"H2494_03389"</span>, <span class="st">"Paraburkholderia"</span>,   <span class="st">"kururiensis"</span>,</span>
<span id="cb15-77"><a href="#cb15-77"></a>  <span class="st">"HAMBI_2659"</span>,          <span class="dv">4</span>L,  <span class="st">"H2659_00367"</span>, <span class="st">"Stenotrophomonas"</span>,   <span class="st">"maltophilia"</span>,</span>
<span id="cb15-78"><a href="#cb15-78"></a>  <span class="st">"HAMBI_2792"</span>,          <span class="dv">4</span>L,  <span class="st">"H2792_00549"</span>,        <span class="st">"Moraxella"</span>,         <span class="st">"canis"</span>,</span>
<span id="cb15-79"><a href="#cb15-79"></a>  <span class="st">"HAMBI_3031"</span>,          <span class="dv">2</span>L,  <span class="st">"H3031_00830"</span>,         <span class="st">"Niabella"</span>,  <span class="st">"yanshanensis"</span>,</span>
<span id="cb15-80"><a href="#cb15-80"></a>  <span class="st">"HAMBI_3237"</span>,          <span class="dv">6</span>L,  <span class="st">"H3237_00875"</span>,       <span class="st">"Microvirga"</span>,   <span class="st">"lotononidis"</span>,</span>
<span id="cb15-81"><a href="#cb15-81"></a>  <span class="st">"HAMBI_1923"</span>,          <span class="dv">6</span>L,  <span class="st">"H1923_00876"</span>,   <span class="st">"Flavobacterium"</span>,      <span class="st">"odoratum"</span></span>
<span id="cb15-82"><a href="#cb15-82"></a>  )</span>
<span id="cb15-83"><a href="#cb15-83"></a><span class="in">```</span></span>
<span id="cb15-84"><a href="#cb15-84"></a></span>
<span id="cb15-85"><a href="#cb15-85"></a><span class="fu">## Functions</span></span>
<span id="cb15-86"><a href="#cb15-86"></a></span>
<span id="cb15-87"><a href="#cb15-87"></a>These are functions to aid in reading and processing output from Rbec</span>
<span id="cb15-88"><a href="#cb15-88"></a></span>
<span id="cb15-91"><a href="#cb15-91"></a><span class="in">```{r}</span></span>
<span id="cb15-92"><a href="#cb15-92"></a><span class="co"># this function </span></span>
<span id="cb15-93"><a href="#cb15-93"></a>normalize_by_copy <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum){</span>
<span id="cb15-94"><a href="#cb15-94"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb15-95"><a href="#cb15-95"></a>    <span class="co"># join with the copy number data frame. We join by the locus tag so this will add H1279_03957 to HAMBI_1299</span></span>
<span id="cb15-96"><a href="#cb15-96"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(tlc, <span class="at">by =</span> <span class="fu">join_by</span>(rRNA16S_locus), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-97"><a href="#cb15-97"></a>    <span class="co"># get total number of mapping reads per species. This aggregates all the difference ASVs per species</span></span>
<span id="cb15-98"><a href="#cb15-98"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(count), <span class="at">.by =</span> <span class="fu">c</span>(sample, strainID, rRNA16S_cn)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-99"><a href="#cb15-99"></a>    <span class="co"># group by sample</span></span>
<span id="cb15-100"><a href="#cb15-100"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb15-101"><a href="#cb15-101"></a>    <span class="co"># calculate a corrected count which is simply the count divided by copy num for each species</span></span>
<span id="cb15-102"><a href="#cb15-102"></a>    <span class="co"># dividide by the sum of count divided by copy num for whole sample multiplied by the total</span></span>
<span id="cb15-103"><a href="#cb15-103"></a>    <span class="co"># number of mapped reads per sample</span></span>
<span id="cb15-104"><a href="#cb15-104"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count_correct =</span> <span class="fu">round</span>(<span class="fu">sum</span>(count)<span class="sc">*</span>(count<span class="sc">/</span>rRNA16S_cn)<span class="sc">/</span><span class="fu">sum</span>(count<span class="sc">/</span>rRNA16S_cn))) <span class="sc">%&gt;%</span>  </span>
<span id="cb15-105"><a href="#cb15-105"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb15-106"><a href="#cb15-106"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count, count_correct)</span>
<span id="cb15-107"><a href="#cb15-107"></a>  }</span>
<span id="cb15-108"><a href="#cb15-108"></a></span>
<span id="cb15-109"><a href="#cb15-109"></a><span class="co"># this function replaces missing species counts with zero</span></span>
<span id="cb15-110"><a href="#cb15-110"></a>completecombos <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum, <span class="at">countname =</span> count, <span class="at">remove1923 =</span> <span class="cn">TRUE</span>){</span>
<span id="cb15-111"><a href="#cb15-111"></a> </span>
<span id="cb15-112"><a href="#cb15-112"></a>  <span class="co"># get unique strainIDs</span></span>
<span id="cb15-113"><a href="#cb15-113"></a>  strainID <span class="ot">&lt;-</span> <span class="fu">unique</span>(tlc<span class="sc">$</span>strainID)</span>
<span id="cb15-114"><a href="#cb15-114"></a>  <span class="co"># table for assigning genus and species names. Doesn't matter if 1923 is there or not</span></span>
<span id="cb15-115"><a href="#cb15-115"></a>  <span class="co"># because it is filter joined later</span></span>
<span id="cb15-116"><a href="#cb15-116"></a>  tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species))</span>
<span id="cb15-117"><a href="#cb15-117"></a>  <span class="cf">if</span> (remove1923) {</span>
<span id="cb15-118"><a href="#cb15-118"></a>    <span class="co"># get unique strainIDs but exclude 1923 if remove1923 is true</span></span>
<span id="cb15-119"><a href="#cb15-119"></a>    strainID <span class="ot">&lt;-</span> strainID[strainID <span class="sc">!=</span> <span class="st">"HAMBI_1923"</span>]</span>
<span id="cb15-120"><a href="#cb15-120"></a>  }</span>
<span id="cb15-121"><a href="#cb15-121"></a>  </span>
<span id="cb15-122"><a href="#cb15-122"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">strainID =</span> strainID, <span class="at">sample =</span> <span class="st">"dummy"</span>), .data) <span class="sc">%&gt;%</span> </span>
<span id="cb15-123"><a href="#cb15-123"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"dummy"</span>, <span class="dv">1</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb15-124"><a href="#cb15-124"></a>    tidyr<span class="sc">::</span><span class="fu">complete</span>(sample, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb15-125"><a href="#cb15-125"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample <span class="sc">!=</span> <span class="st">"dummy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-126"><a href="#cb15-126"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(<span class="fu">is.na</span>({{ countname }}), <span class="dv">0</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb15-127"><a href="#cb15-127"></a>    tidyr<span class="sc">::</span><span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">count_correct =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-128"><a href="#cb15-128"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species)), <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-129"><a href="#cb15-129"></a>    dplyr<span class="sc">::</span><span class="fu">relocate</span>(genus, species, <span class="at">.after =</span> strainID)</span>
<span id="cb15-130"><a href="#cb15-130"></a>}</span>
<span id="cb15-131"><a href="#cb15-131"></a><span class="in">```</span></span>
<span id="cb15-132"><a href="#cb15-132"></a></span>
<span id="cb15-133"><a href="#cb15-133"></a><span class="fu"># Read data</span></span>
<span id="cb15-134"><a href="#cb15-134"></a></span>
<span id="cb15-135"><a href="#cb15-135"></a><span class="fu">## Metadata</span></span>
<span id="cb15-136"><a href="#cb15-136"></a></span>
<span id="cb15-137"><a href="#cb15-137"></a>Reads metadata for the samples</span>
<span id="cb15-138"><a href="#cb15-138"></a></span>
<span id="cb15-141"><a href="#cb15-141"></a><span class="in">```{r}</span></span>
<span id="cb15-142"><a href="#cb15-142"></a><span class="co">#| output: false</span></span>
<span id="cb15-143"><a href="#cb15-143"></a><span class="co">#| warning: false</span></span>
<span id="cb15-144"><a href="#cb15-144"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240318_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb15-145"><a href="#cb15-145"></a><span class="in">```</span></span>
<span id="cb15-146"><a href="#cb15-146"></a></span>
<span id="cb15-147"><a href="#cb15-147"></a><span class="fu">## Rbec </span></span>
<span id="cb15-148"><a href="#cb15-148"></a></span>
<span id="cb15-149"><a href="#cb15-149"></a>Untar Rbec output tarball which contains raw counts tables</span>
<span id="cb15-150"><a href="#cb15-150"></a></span>
<span id="cb15-153"><a href="#cb15-153"></a><span class="in">```{r}</span></span>
<span id="cb15-154"><a href="#cb15-154"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb15-155"><a href="#cb15-155"></a>  amplicontar,</span>
<span id="cb15-156"><a href="#cb15-156"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb15-157"><a href="#cb15-157"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb15-158"><a href="#cb15-158"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb15-159"><a href="#cb15-159"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb15-160"><a href="#cb15-160"></a>)</span>
<span id="cb15-161"><a href="#cb15-161"></a><span class="in">```</span></span>
<span id="cb15-162"><a href="#cb15-162"></a></span>
<span id="cb15-163"><a href="#cb15-163"></a>Sets up directory structure for decompressing</span>
<span id="cb15-164"><a href="#cb15-164"></a></span>
<span id="cb15-167"><a href="#cb15-167"></a><span class="in">```{r}</span></span>
<span id="cb15-168"><a href="#cb15-168"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb15-169"><a href="#cb15-169"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb15-170"><a href="#cb15-170"></a>sampnames <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb15-171"><a href="#cb15-171"></a>  purrr<span class="sc">::</span><span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span>
<span id="cb15-172"><a href="#cb15-172"></a><span class="in">```</span></span>
<span id="cb15-173"><a href="#cb15-173"></a></span>
<span id="cb15-174"><a href="#cb15-174"></a>Reads all the <span class="in">`strain_table.txt`</span> files in a loop</span>
<span id="cb15-175"><a href="#cb15-175"></a></span>
<span id="cb15-178"><a href="#cb15-178"></a><span class="in">```{r}</span></span>
<span id="cb15-179"><a href="#cb15-179"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-180"><a href="#cb15-180"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb15-181"><a href="#cb15-181"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(</span>
<span id="cb15-182"><a href="#cb15-182"></a>  readr<span class="sc">::</span>read_tsv,</span>
<span id="cb15-183"><a href="#cb15-183"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb15-184"><a href="#cb15-184"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb15-185"><a href="#cb15-185"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-186"><a href="#cb15-186"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>)</span>
<span id="cb15-187"><a href="#cb15-187"></a><span class="in">```</span></span>
<span id="cb15-188"><a href="#cb15-188"></a></span>
<span id="cb15-189"><a href="#cb15-189"></a><span class="fu"># Format</span></span>
<span id="cb15-190"><a href="#cb15-190"></a></span>
<span id="cb15-191"><a href="#cb15-191"></a>Calls function from above that normalize counts by 16S copy number</span>
<span id="cb15-192"><a href="#cb15-192"></a></span>
<span id="cb15-195"><a href="#cb15-195"></a><span class="in">```{r}</span></span>
<span id="cb15-196"><a href="#cb15-196"></a><span class="co">#| output: false</span></span>
<span id="cb15-197"><a href="#cb15-197"></a><span class="co">#| warning: false</span></span>
<span id="cb15-198"><a href="#cb15-198"></a>straintabs_norm <span class="ot">&lt;-</span> <span class="fu">normalize_by_copy</span>(straintabs)</span>
<span id="cb15-199"><a href="#cb15-199"></a><span class="in">```</span></span>
<span id="cb15-200"><a href="#cb15-200"></a></span>
<span id="cb15-201"><a href="#cb15-201"></a>Calls function from above that complete all combinations of 23 species</span>
<span id="cb15-202"><a href="#cb15-202"></a></span>
<span id="cb15-205"><a href="#cb15-205"></a><span class="in">```{r}</span></span>
<span id="cb15-206"><a href="#cb15-206"></a>straintabs_norm_fmt <span class="ot">&lt;-</span> <span class="fu">completecombos</span>(straintabs_norm, <span class="at">tlc =</span> tax_locus_copynum, <span class="at">countname =</span> count, <span class="at">remove1923 =</span> <span class="cn">TRUE</span>) </span>
<span id="cb15-207"><a href="#cb15-207"></a><span class="in">```</span></span>
<span id="cb15-208"><a href="#cb15-208"></a></span>
<span id="cb15-209"><a href="#cb15-209"></a>Joins the count data and metadata</span>
<span id="cb15-210"><a href="#cb15-210"></a></span>
<span id="cb15-213"><a href="#cb15-213"></a><span class="in">```{r}</span></span>
<span id="cb15-214"><a href="#cb15-214"></a>finaltable <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(straintabs_norm_fmt, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample))</span>
<span id="cb15-215"><a href="#cb15-215"></a><span class="in">```</span></span>
<span id="cb15-216"><a href="#cb15-216"></a></span>
<span id="cb15-217"><a href="#cb15-217"></a>Parition out the two different projects (tipping points and pleiotropy were sequenced on same flow cell) and filter the pleiotropy samples to only include samples for ancestral community, YSK bacteria, and YSK streptomycin</span>
<span id="cb15-218"><a href="#cb15-218"></a></span>
<span id="cb15-221"><a href="#cb15-221"></a><span class="in">```{r}</span></span>
<span id="cb15-222"><a href="#cb15-222"></a>finaltip <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb15-223"><a href="#cb15-223"></a>  <span class="fu">filter</span>(project <span class="sc">==</span> <span class="st">"tipping_points"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-224"><a href="#cb15-224"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pool<span class="sc">:</span>project), <span class="sc">-</span>lib_prep, <span class="sc">-</span>measure_env, <span class="sc">-</span>evolution_env)</span>
<span id="cb15-225"><a href="#cb15-225"></a></span>
<span id="cb15-226"><a href="#cb15-226"></a>finalpleio <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb15-227"><a href="#cb15-227"></a>  <span class="fu">filter</span>(project <span class="sc">==</span> <span class="st">"amanda_chapt01"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-228"><a href="#cb15-228"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pool<span class="sc">:</span>project), <span class="sc">-</span>lib_prep, <span class="sc">-</span>evo_hist, <span class="sc">-</span>amp_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb15-229"><a href="#cb15-229"></a>  <span class="fu">filter</span>(measure_env <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"none"</span>, <span class="st">"bact"</span>, <span class="st">"bact_strep"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-230"><a href="#cb15-230"></a>  <span class="fu">filter</span>(evolution_env <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"bact"</span>, <span class="st">"bact_strep"</span>))</span>
<span id="cb15-231"><a href="#cb15-231"></a><span class="in">```</span></span>
<span id="cb15-232"><a href="#cb15-232"></a></span>
<span id="cb15-233"><a href="#cb15-233"></a><span class="fu"># Write results</span></span>
<span id="cb15-234"><a href="#cb15-234"></a></span>
<span id="cb15-235"><a href="#cb15-235"></a>Write table of species abundances with metadata</span>
<span id="cb15-236"><a href="#cb15-236"></a></span>
<span id="cb15-239"><a href="#cb15-239"></a><span class="in">```{r}</span></span>
<span id="cb15-240"><a href="#cb15-240"></a>readr<span class="sc">:::</span><span class="fu">write_tsv</span>(finalpleio, here<span class="sc">::</span><span class="fu">here</span>(data_sp, <span class="st">"species_counts_md.tsv"</span>))</span>
<span id="cb15-241"><a href="#cb15-241"></a><span class="in">```</span></span>
<span id="cb15-242"><a href="#cb15-242"></a></span>
<span id="cb15-243"><a href="#cb15-243"></a><span class="fu"># Clean up</span></span>
<span id="cb15-244"><a href="#cb15-244"></a></span>
<span id="cb15-245"><a href="#cb15-245"></a>Remove decompressed coverage directory from temp location</span>
<span id="cb15-248"><a href="#cb15-248"></a><span class="in">```{r}</span></span>
<span id="cb15-249"><a href="#cb15-249"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span>
<span id="cb15-250"><a href="#cb15-250"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>