<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-11-03">

<title>hambiYSKpleiotropy - Analysis of community compositions</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<script src="../../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">2. Community composition workflow (STR)</a></li><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/02_composition_analysis.html">ii) Community composition</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../../">hambiYSKpleiotropy</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">1. Community densities workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/cell_density/01_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot densities</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">2. Community composition workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Amplicon data wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/amplicon_sp_counts/02_composition_analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">ii) Community composition</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/communities/machine_learning_pipelines/01_machine_learn_home_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Machine learning the home environment</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">3. Clones workflow (STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/01_growth_curve_qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect/Smooth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/02_growthrate_auc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Summarize curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_only/monocultures/clone_phenotyping/03_growth_summary_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Analysis and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">1A. Community densities workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/communities/cell_density/01_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot densities</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">2A. Community composition workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/communities/amplicon_sp_counts/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Amplicon data wrangling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/communities/amplicon_sp_counts/02_composition_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Community composition</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/communities/machine_learning_pipelines/01_machine_learn_home_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Machine learning the home environment</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true">
 <span class="menu-text">3A. Clones workflow (Predator and STR)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/monocultures/clone_phenotyping/01_growth_curve_qc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect/Smooth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/monocultures/clone_phenotyping/02_growthrate_auc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Summarize curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/strep_predator/monocultures/clone_phenotyping/03_growth_summary_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Analysis and plot</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read" id="toc-read" class="nav-link" data-scroll-target="#read"><span class="header-section-number">2</span> Read</a></li>
  <li><a href="#formatting" id="toc-formatting" class="nav-link" data-scroll-target="#formatting"><span class="header-section-number">3</span> Formatting</a></li>
  <li><a href="#visualize-community-composition" id="toc-visualize-community-composition" class="nav-link" data-scroll-target="#visualize-community-composition"><span class="header-section-number">4</span> Visualize community composition</a></li>
  <li><a href="#tsne-dimension-reduction-of-community-composition" id="toc-tsne-dimension-reduction-of-community-composition" class="nav-link" data-scroll-target="#tsne-dimension-reduction-of-community-composition"><span class="header-section-number">5</span> tSNE dimension reduction of community composition</a></li>
  <li><a href="#distances-between-communities" id="toc-distances-between-communities" class="nav-link" data-scroll-target="#distances-between-communities"><span class="header-section-number">6</span> Distances between communities</a>
  <ul class="collapse">
  <li><a href="#jsd-to-ancestor-no-samples-filtered" id="toc-jsd-to-ancestor-no-samples-filtered" class="nav-link" data-scroll-target="#jsd-to-ancestor-no-samples-filtered"><span class="header-section-number">6.1</span> JSD to ancestor no samples filtered</a></li>
  <li><a href="#jsd-to-ancestor---idiosyncratic-samples-filtered" id="toc-jsd-to-ancestor---idiosyncratic-samples-filtered" class="nav-link" data-scroll-target="#jsd-to-ancestor---idiosyncratic-samples-filtered"><span class="header-section-number">6.2</span> JSD to ancestor - idiosyncratic samples filtered</a></li>
  <li><a href="#jsd-to-ysk-founder-community" id="toc-jsd-to-ysk-founder-community" class="nav-link" data-scroll-target="#jsd-to-ysk-founder-community"><span class="header-section-number">6.3</span> JSD to YSK founder community</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/01_format_rbec_tab.html">2. Community composition workflow (STR)</a></li><li class="breadcrumb-item"><a href="../../../../R/strep_only/communities/amplicon_sp_counts/02_composition_analysis.html">ii) Community composition</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Analysis of community compositions</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
<p class="subtitle lead">From the STR community composition workflow</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook plots community composition of the experiment, conducts ordination of CLR transformed count data, and compares the response of different YSK evolutionary histories to the ancestral community in each experimental measurement environment using pairwise Jensen-Shannon divergences.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Loads required libraries and sets global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(Rtsne)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(philentropy)</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read</h1>
<p>Read species abundance data (16S v3 amplicon counts) and do some light formatting of metadata</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>sptable <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_sp, <span class="st">"species_counts_md.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-2"><a href="#cb2-2"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> day<span class="sc">/</span><span class="dv">7</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a>counts_f <span class="ot">&lt;-</span> sptable <span class="sc">%&gt;%</span> </span>
<span id="cb2-5"><a href="#cb2-5"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb2-6"><a href="#cb2-6"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">f=</span>count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-7"><a href="#cb2-7"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-8"><a href="#cb2-8"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">measure_env_short =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(measure_env <span class="sc">==</span> <span class="st">"bact"</span> <span class="sc">~</span> <span class="st">"Meas: B"</span>,</span>
<span id="cb2-9"><a href="#cb2-9"></a>                                       measure_env <span class="sc">==</span> <span class="st">"bact_strep"</span> <span class="sc">~</span> <span class="st">"Meas: BS"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-10"><a href="#cb2-10"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb2-11"><a href="#cb2-11"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"Home: Anc"</span>,</span>
<span id="cb2-12"><a href="#cb2-12"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"bact"</span> <span class="sc">~</span> <span class="st">"Home: B"</span>,</span>
<span id="cb2-13"><a href="#cb2-13"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"bact_strep"</span> <span class="sc">~</span> <span class="st">"Home: BS"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-14"><a href="#cb2-14"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-15"><a href="#cb2-15"></a>    <span class="at">measure_env_short =</span> <span class="fu">factor</span>(measure_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Meas: B"</span>, <span class="st">"Meas: BS"</span>)),</span>
<span id="cb2-16"><a href="#cb2-16"></a>    <span class="at">home_env_short =</span> <span class="fu">factor</span>(home_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Home: Anc"</span>, <span class="st">"Home: B"</span>, <span class="st">"Home: BS"</span>)),</span>
<span id="cb2-17"><a href="#cb2-17"></a>         <span class="at">day =</span> <span class="fu">factor</span>(day),</span>
<span id="cb2-18"><a href="#cb2-18"></a>         <span class="at">replicate =</span> <span class="fu">factor</span>(replicate),</span>
<span id="cb2-19"><a href="#cb2-19"></a>         <span class="at">strainID =</span> <span class="fu">factor</span>(strainID, <span class="at">levels =</span> <span class="fu">names</span>(hambi_colors)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="formatting" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Formatting</h1>
<p>Some light formatting to subset data into distint tibbles for later plotting</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># these are communities of a (supposedly) known composition. Can be used with metacal</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>pos_ctrl_samples <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(sample, <span class="st">"pos_ctrl"</span>))</span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># these are samples taken directly from YSK and represent the composition of the communities used to start the experiment</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>t0_samples <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb3-7"><a href="#cb3-7"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(sample, <span class="st">"pos_ctrl"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-8"><a href="#cb3-8"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(day <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb3-9"><a href="#cb3-9"></a></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co"># only samples from the experiment</span></span>
<span id="cb3-11"><a href="#cb3-11"></a>counts_f_experiment <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">anti_join</span>(counts_f, pos_ctrl_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb3-12"><a href="#cb3-12"></a>  dplyr<span class="sc">::</span><span class="fu">anti_join</span>(., t0_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb3-13"><a href="#cb3-13"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-14"><a href="#cb3-14"></a>    <span class="at">measure_env_short =</span> <span class="fu">factor</span>(measure_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Meas: B"</span>, <span class="st">"Meas: BS"</span>)),</span>
<span id="cb3-15"><a href="#cb3-15"></a>    <span class="at">home_env_short =</span> <span class="fu">factor</span>(home_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Home: Anc"</span>, <span class="st">"Home: B"</span>, <span class="st">"Home: BS"</span>)),</span>
<span id="cb3-16"><a href="#cb3-16"></a>         <span class="at">day =</span> <span class="fu">factor</span>(day),</span>
<span id="cb3-17"><a href="#cb3-17"></a>         <span class="at">replicate =</span> <span class="fu">factor</span>(replicate),</span>
<span id="cb3-18"><a href="#cb3-18"></a>         <span class="at">strainID =</span> <span class="fu">factor</span>(strainID, <span class="at">levels =</span> <span class="fu">names</span>(hambi_colors)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="visualize-community-composition" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Visualize community composition</h1>
<p>Using ggplot2 to plot bar plots of species composition from the experiment and the intial material used to seed the experiment</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># custom barplot theme</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>mybartheme <span class="ot">&lt;-</span> <span class="cf">function</span>(...){</span>
<span id="cb4-3"><a href="#cb4-3"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb4-4"><a href="#cb4-4"></a>    <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>,<span class="st">"line"</span>),</span>
<span id="cb4-5"><a href="#cb4-5"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-7"><a href="#cb4-7"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-8"><a href="#cb4-8"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-9"><a href="#cb4-9"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-10"><a href="#cb4-10"></a>    <span class="co">#axis.text.x = element_blank(),</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb4-12"><a href="#cb4-12"></a>    <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb4-13"><a href="#cb4-13"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-14"><a href="#cb4-14"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-15"><a href="#cb4-15"></a>    <span class="at">legend.key =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-16"><a href="#cb4-16"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb4-17"><a href="#cb4-17"></a>    ...)</span>
<span id="cb4-18"><a href="#cb4-18"></a>}</span>
<span id="cb4-19"><a href="#cb4-19"></a></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="co"># Base function for making the subplots</span></span>
<span id="cb4-21"><a href="#cb4-21"></a>basebar <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">x =</span> x, <span class="at">xlab =</span> xlab, <span class="at">xgrid =</span> <span class="cn">NULL</span>){</span>
<span id="cb4-22"><a href="#cb4-22"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(.data) <span class="sc">+</span></span>
<span id="cb4-23"><a href="#cb4-23"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> f, <span class="at">x =</span> {{ x }}, <span class="at">fill =</span> strainID), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">linewidth=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb4-24"><a href="#cb4-24"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> xlab, <span class="at">y=</span><span class="st">"% abundance"</span>, <span class="at">fill=</span><span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb4-25"><a href="#cb4-25"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb4-26"><a href="#cb4-26"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb4-27"><a href="#cb4-27"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> ggh4x<span class="sc">::</span><span class="fu">guide_axis_nested</span>()) <span class="sc">+</span></span>
<span id="cb4-28"><a href="#cb4-28"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb4-29"><a href="#cb4-29"></a>  <span class="fu">mybartheme</span>()</span>
<span id="cb4-30"><a href="#cb4-30"></a>}</span>
<span id="cb4-31"><a href="#cb4-31"></a></span>
<span id="cb4-32"><a href="#cb4-32"></a>A <span class="ot">&lt;-</span> t0_samples <span class="sc">%&gt;%</span> </span>
<span id="cb4-33"><a href="#cb4-33"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-34"><a href="#cb4-34"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> replicate, <span class="at">xlab =</span> <span class="st">"Replicate"</span>) <span class="sc">+</span></span>
<span id="cb4-35"><a href="#cb4-35"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>( <span class="sc">~</span> home_env_short)</span>
<span id="cb4-36"><a href="#cb4-36"></a></span>
<span id="cb4-37"><a href="#cb4-37"></a>B <span class="ot">&lt;-</span> t0_samples <span class="sc">%&gt;%</span> </span>
<span id="cb4-38"><a href="#cb4-38"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">!=</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-39"><a href="#cb4-39"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> replicate, <span class="at">xlab =</span> <span class="st">"Replicate"</span>) <span class="sc">+</span></span>
<span id="cb4-40"><a href="#cb4-40"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>( <span class="sc">~</span> home_env_short)</span>
<span id="cb4-41"><a href="#cb4-41"></a></span>
<span id="cb4-42"><a href="#cb4-42"></a>C <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb4-43"><a href="#cb4-43"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-44"><a href="#cb4-44"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> <span class="fu">interaction</span>(replicate, transfer), <span class="at">xlab =</span> <span class="st">"Replicate | Growth Cycle"</span>) <span class="sc">+</span></span>
<span id="cb4-45"><a href="#cb4-45"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short <span class="sc">~</span> home_env_short)</span>
<span id="cb4-46"><a href="#cb4-46"></a></span>
<span id="cb4-47"><a href="#cb4-47"></a>D <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb4-48"><a href="#cb4-48"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">!=</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-49"><a href="#cb4-49"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> <span class="fu">interaction</span>(replicate, transfer), <span class="at">xlab =</span> <span class="st">"Replicate | Growth Cycle"</span>) <span class="sc">+</span></span>
<span id="cb4-50"><a href="#cb4-50"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short <span class="sc">~</span> home_env_short)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Makes a final patchwork object and saves the results as raster and vector</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>layout <span class="ot">&lt;-</span> <span class="st">"</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="st">ABB</span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="st">CDD</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="st">"</span></span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a>pbar <span class="ot">&lt;-</span> A <span class="sc">+</span> B <span class="sc">+</span> C <span class="sc">+</span> D  <span class="sc">+</span></span>
<span id="cb5-7"><a href="#cb5-7"></a>  patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.4</span>), </span>
<span id="cb5-8"><a href="#cb5-8"></a>              <span class="at">design =</span> layout) <span class="sc">+</span></span>
<span id="cb5-9"><a href="#cb5-9"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">'bottom'</span>)</span>
<span id="cb5-11"><a href="#cb5-11"></a></span>
<span id="cb5-12"><a href="#cb5-12"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb5-13"><a href="#cb5-13"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_bar_plot.svg"</span>),</span>
<span id="cb5-14"><a href="#cb5-14"></a>  pbar,</span>
<span id="cb5-15"><a href="#cb5-15"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb5-16"><a href="#cb5-16"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb5-17"><a href="#cb5-17"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb5-18"><a href="#cb5-18"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb5-19"><a href="#cb5-19"></a>)</span>
<span id="cb5-20"><a href="#cb5-20"></a></span>
<span id="cb5-21"><a href="#cb5-21"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb5-22"><a href="#cb5-22"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_bar_plot.png"</span>),</span>
<span id="cb5-23"><a href="#cb5-23"></a>  pbar,</span>
<span id="cb5-24"><a href="#cb5-24"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb5-25"><a href="#cb5-25"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb5-26"><a href="#cb5-26"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb5-27"><a href="#cb5-27"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb5-28"><a href="#cb5-28"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_composition_analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Relative abundance (Y axis, percent) of species from replicate microcosms (X axis, A-F). <strong>A)</strong> Community composition of the ancestral/clonal replicates inocula used to start experiments in C. <strong>B)</strong> Composition of the long-term community evolution experiment (YSK, week X) communities from each selection condition. These communities were used to inoculate the transplantation experiment in part D. <strong>C)</strong> Composition of the ancestral/clonal replicate communities after 4, 8, and 12 transfers in each measurement environment (grid rows). <strong>D)</strong> Composition of the YSK communities (grid columns, “home” environment) subjected to all possible measurement conditions (grid rows). B = bacteria grown without amendment, BS = streptomycin treatment, BP = <em>Tetrahymena thermophila</em> treatment, BPS = combined <em>Tetrahymena thermophila</em> and streptomycin treatment.
</figcaption>
</figure>
</div>
</section>
<section id="tsne-dimension-reduction-of-community-composition" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> tSNE dimension reduction of community composition</h1>
<p>Some preliminary inspections of the number of samples with zeros. This is needed for zcompositions</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">n_samples =</span> <span class="fu">n</span>(),</span>
<span id="cb6-4"><a href="#cb6-4"></a>            <span class="at">n_gt0 =</span> <span class="fu">sum</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb6-5"><a href="#cb6-5"></a>            <span class="at">p_gt0 =</span> n_gt0 <span class="sc">/</span> n_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(n_gt0)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["strainID"],"name":[1],"type":["fct"],"align":["left"]},{"label":["n_samples"],"name":[2],"type":["int"],"align":["right"]},{"label":["n_gt0"],"name":[3],"type":["int"],"align":["right"]},{"label":["p_gt0"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"HAMBI_2792","2":"107","3":"0","4":"0.000000000"},{"1":"HAMBI_0097","2":"107","3":"1","4":"0.009345794"},{"1":"HAMBI_0262","2":"107","3":"2","4":"0.018691589"},{"1":"HAMBI_1988","2":"107","3":"6","4":"0.056074766"},{"1":"HAMBI_1299","2":"107","3":"25","4":"0.233644860"},{"1":"HAMBI_3237","2":"107","3":"29","4":"0.271028037"},{"1":"HAMBI_1279","2":"107","3":"30","4":"0.280373832"},{"1":"HAMBI_2159","2":"107","3":"67","4":"0.626168224"},{"1":"HAMBI_2443","2":"107","3":"83","4":"0.775700935"},{"1":"HAMBI_2494","2":"107","3":"84","4":"0.785046729"},{"1":"HAMBI_0006","2":"107","3":"95","4":"0.887850467"},{"1":"HAMBI_2164","2":"107","3":"99","4":"0.925233645"},{"1":"HAMBI_1972","2":"107","3":"103","4":"0.962616822"},{"1":"HAMBI_1292","2":"107","3":"106","4":"0.990654206"},{"1":"HAMBI_3031","2":"107","3":"106","4":"0.990654206"},{"1":"HAMBI_2160","2":"107","3":"106","4":"0.990654206"},{"1":"HAMBI_1287","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_1977","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_0403","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_2659","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_1896","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_0105","2":"107","3":"107","4":"1.000000000"},{"1":"HAMBI_1842","2":"107","3":"107","4":"1.000000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Here we will use the <a href="https://en.wikipedia.org/wiki/Compositional_data#Center_log_ratio_transform">centered log-ratio transformation</a> for the species abundances. The centered log-ratio can be interpreted as the log-fold change of species i relative to the average microbe in a sample. The formula for the transformation is:</p>
<p><span id="eq-clr"><span class="math display">\[
\text{clr}(\mathbf x)= \left(log
\frac{x_i}{g(\mathbf x)} \right)_{i=1,...,D} \qquad \text{with} \quad
g(\mathbf x) = \left(\prod_{i=1}^Dx_i\right)^{1/D} =
\exp\left(\frac{1}{D}\sum_{i=1}^D \log x_i\right)\text{,}
\tag{1}\]</span></span></p>
<p>We will use the implementation of centered log-ratio transform in the <code>compositions</code> package</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">set.seed</span>(<span class="dv">234781</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co"># exclude these species because they have too many zeros</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>lowstrainsv <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="st">"HAMBI_0097"</span>,</span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="st">"HAMBI_2792"</span></span>
<span id="cb7-7"><a href="#cb7-7"></a>)</span>
<span id="cb7-8"><a href="#cb7-8"></a></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co"># make a species count matrix</span></span>
<span id="cb7-10"><a href="#cb7-10"></a>mymat <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb7-11"><a href="#cb7-11"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(strainID <span class="sc">%in%</span> lowstrainsv)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-12"><a href="#cb7-12"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count) <span class="sc">%&gt;%</span> </span>
<span id="cb7-13"><a href="#cb7-13"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count =</span> count <span class="sc">+</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="co"># important to arrange by sample as this makes some later joins easier</span></span>
<span id="cb7-15"><a href="#cb7-15"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb7-16"><a href="#cb7-16"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"strainID"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-17"><a href="#cb7-17"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-18"><a href="#cb7-18"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb7-19"><a href="#cb7-19"></a></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co"># calculate clr</span></span>
<span id="cb7-21"><a href="#cb7-21"></a>balclr <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(mymat)</span>
<span id="cb7-22"><a href="#cb7-22"></a></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co"># tsne</span></span>
<span id="cb7-24"><a href="#cb7-24"></a>mymat_tsne <span class="ot">&lt;-</span> Rtsne<span class="sc">::</span><span class="fu">Rtsne</span>(<span class="fu">as.matrix</span>(balclr), <span class="at">perplexity =</span> <span class="dv">30</span>, <span class="at">dims =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Define a plotting function to plot the tSNE output</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>ptsne <span class="ot">&lt;-</span> <span class="cf">function</span>(tsne_obj, colorvar, shapevar, colors){</span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="fu">data.frame</span>(tsne_obj<span class="sc">$</span>Y) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">cbind</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(counts_f_experiment, sample, replicate, transfer, measure_env_short, home_env_short))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="fu">factor</span>(transfer)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span> X2, <span class="at">y =</span> X1, <span class="at">color =</span> {{ colorvar }}, <span class="at">shape =</span> {{ shapevar }}), <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8"></a>  ggplot2<span class="sc">::</span><span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb8-9"><a href="#cb8-9"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"tSNE dimension 1"</span>, </span>
<span id="cb8-10"><a href="#cb8-10"></a>       <span class="at">y =</span> <span class="st">"tSNE dimension 2"</span>, </span>
<span id="cb8-11"><a href="#cb8-11"></a>       <span class="at">shape =</span> <span class="st">"Transfer"</span>) <span class="sc">+</span></span>
<span id="cb8-12"><a href="#cb8-12"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> colors) <span class="sc">+</span></span>
<span id="cb8-13"><a href="#cb8-13"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb8-14"><a href="#cb8-14"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb8-15"><a href="#cb8-15"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-16"><a href="#cb8-16"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-17"><a href="#cb8-17"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-18"><a href="#cb8-18"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb8-19"><a href="#cb8-19"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-20"><a href="#cb8-20"></a>  ) </span>
<span id="cb8-21"><a href="#cb8-21"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Call the plotting function, make the patchwork layout, and save in raster and vector format</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>pm <span class="ot">&lt;-</span> <span class="fu">ptsne</span>(mymat_tsne, measure_env_short, <span class="at">shapevar =</span> transfer, <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>)) <span class="sc">+</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Measure</span><span class="sc">\n</span><span class="st">environment"</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>ph <span class="ot">&lt;-</span> <span class="fu">ptsne</span>(mymat_tsne, home_env_short, <span class="at">shapevar =</span> transfer, <span class="fu">c</span>(<span class="st">"#8da0cb"</span>, <span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>)) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">environment"</span>)</span>
<span id="cb9-5"><a href="#cb9-5"></a></span>
<span id="cb9-6"><a href="#cb9-6"></a>pt <span class="ot">&lt;-</span> pm <span class="sc">+</span> ph <span class="sc">+</span></span>
<span id="cb9-7"><a href="#cb9-7"></a>  patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>)</span>
<span id="cb9-9"><a href="#cb9-9"></a></span>
<span id="cb9-10"><a href="#cb9-10"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb9-11"><a href="#cb9-11"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_tsne.svg"</span>),</span>
<span id="cb9-12"><a href="#cb9-12"></a>  pt,</span>
<span id="cb9-13"><a href="#cb9-13"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb9-14"><a href="#cb9-14"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb9-15"><a href="#cb9-15"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb9-16"><a href="#cb9-16"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb9-17"><a href="#cb9-17"></a>)</span>
<span id="cb9-18"><a href="#cb9-18"></a></span>
<span id="cb9-19"><a href="#cb9-19"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb9-20"><a href="#cb9-20"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_tsne.png"</span>),</span>
<span id="cb9-21"><a href="#cb9-21"></a>  pt,</span>
<span id="cb9-22"><a href="#cb9-22"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb9-23"><a href="#cb9-23"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb9-24"><a href="#cb9-24"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb9-25"><a href="#cb9-25"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb9-26"><a href="#cb9-26"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_composition_analysis_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Dimensional reduction of bacterial community composition using t-SNE. Each point represents a replicate microcosm sampled at serial transfers 4, 8, and 12 (grid columns). In <strong>A)</strong> point colors depict the experimental measurement condition from the transplantation experiment. In <strong>B)</strong> point colors depict the evolutionary history of the inoculating communities (“home environment”).
</figcaption>
</figure>
</div>
</section>
<section id="distances-between-communities" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Distances between communities</h1>
<p>The Jensen–Shannon divergence (JSD) is a symmetrized and smoothed version of the <a href="https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence" title="Kullback–Leibler divergence">Kullback–Leibler divergence</a>. It is defined as:</p>
<p><span id="eq-JSD"><span class="math display">\[
D_{\text{JS}}(P \parallel Q) = \frac{1}{2} D_{\text{KL}}(P \parallel M) + \frac{1}{2} D_{\text{KL}}(Q \parallel M)
\tag{2}\]</span></span></p>
<p><span id="eq-avg_dist"><span class="math display">\[
M = \frac{1}{2}(P + Q)
\tag{3}\]</span></span></p>
<p>Here, (<span class="math inline">\(D_{\text{KL}}\)</span>) denotes the Kullback-Leibler divergence which is defined as:</p>
<p><span id="eq-KLD"><span class="math display">\[
D_{KL}(P \parallel Q) = \sum_{x \in \mathcal{X}} P(x) \log \frac{P(x)}{Q(x)}
\tag{4}\]</span></span></p>
<section id="jsd-to-ancestor-no-samples-filtered" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="jsd-to-ancestor-no-samples-filtered"><span class="header-section-number">6.1</span> JSD to ancestor no samples filtered</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># create a matrix of species frequencies</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>prob_mat <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> strainID, <span class="at">values_from =</span> f) <span class="sc">%&gt;%</span> </span>
<span id="cb10-5"><a href="#cb10-5"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb10-7"><a href="#cb10-7"></a></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="co"># calculate JSD between all pairs using philentropy::JSD</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>jsd_mat <span class="ot">&lt;-</span> philentropy<span class="sc">::</span><span class="fu">JSD</span>(prob_mat)</span>
<span id="cb10-10"><a href="#cb10-10"></a></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="co"># get nonredundant pairings</span></span>
<span id="cb10-12"><a href="#cb10-12"></a>ind <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">upper.tri</span>(jsd_mat, <span class="at">diag =</span> <span class="cn">TRUE</span>), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-13"><a href="#cb10-13"></a>nn <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">rownames</span>(prob_mat), <span class="fu">rownames</span>(prob_mat))</span>
<span id="cb10-14"><a href="#cb10-14"></a>jsddf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample.x =</span> nn[[<span class="dv">1</span>]][ind[, <span class="dv">1</span>]],</span>
<span id="cb10-15"><a href="#cb10-15"></a>           <span class="at">sample.y =</span> nn[[<span class="dv">2</span>]][ind[, <span class="dv">2</span>]],</span>
<span id="cb10-16"><a href="#cb10-16"></a>           <span class="at">jsd =</span> jsd_mat[ind])</span>
<span id="cb10-17"><a href="#cb10-17"></a></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="co"># metadata for joining</span></span>
<span id="cb10-19"><a href="#cb10-19"></a>md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(counts_f, sample, replicate, transfer, measure_env_short, home_env_short)</span>
<span id="cb10-20"><a href="#cb10-20"></a></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="co"># convert to long format</span></span>
<span id="cb10-22"><a href="#cb10-22"></a>jsddf_long_md <span class="ot">&lt;-</span> jsddf <span class="sc">%&gt;%</span> </span>
<span id="cb10-23"><a href="#cb10-23"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(md, <span class="at">by =</span> <span class="fu">join_by</span>(sample.x<span class="sc">==</span>sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-24"><a href="#cb10-24"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(md, <span class="at">by =</span> <span class="fu">join_by</span>(sample.y<span class="sc">==</span>sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Filter JSD pairs to only those that are relative to ancestral community in each measurement environment. Also calculate mean and boostrapped 95CI using HMisc</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>jsddf_long_md_filt <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Hmmm…. not sure what exactly this is telling, but I am calculating the within home/measurement divergence as some kind of variability baseline.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>intrarepvar <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.y <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Make plot of JSD to ancestral communities in each condition and save in raster/vector format</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>plot_jsd <span class="ot">&lt;-</span> <span class="cf">function</span>(intra_df, df){</span>
<span id="cb13-2"><a href="#cb13-2"></a>  mycols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Home: B"</span> <span class="ot">=</span> <span class="st">"#66c2a5"</span>, <span class="st">"Home: BS"</span> <span class="ot">=</span> <span class="st">"#fc8d62"</span>)</span>
<span id="cb13-3"><a href="#cb13-3"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_ribbon</span>(<span class="at">data =</span> intra_df, </span>
<span id="cb13-5"><a href="#cb13-5"></a>              <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">ymin =</span> ymin, <span class="at">ymax=</span>ymax, <span class="at">fill =</span> home_env_short.y), <span class="at">alpha=</span><span class="fl">0.25</span>) <span class="sc">+</span> </span>
<span id="cb13-6"><a href="#cb13-6"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">data =</span> intra_df, </span>
<span id="cb13-7"><a href="#cb13-7"></a>            <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y=</span>y, <span class="at">color =</span> home_env_short.y), <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">lty=</span><span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb13-8"><a href="#cb13-8"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y =</span> y, <span class="at">group =</span> home_env_short.y, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span></span>
<span id="cb13-9"><a href="#cb13-9"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_linerange</span>(<span class="at">data =</span> df, </span>
<span id="cb13-10"><a href="#cb13-10"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">ymin =</span> ymin, <span class="at">ymax =</span> ymax, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span> </span>
<span id="cb13-11"><a href="#cb13-11"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">data=</span> df, </span>
<span id="cb13-12"><a href="#cb13-12"></a>            <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y =</span> y, <span class="at">group =</span> home_env_short.y, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span> </span>
<span id="cb13-13"><a href="#cb13-13"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short.y <span class="sc">~</span> home_env_short.y, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span> </span>
<span id="cb13-14"><a href="#cb13-14"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Growth Cycles"</span>, <span class="at">y =</span> <span class="st">"Jensen-Shannon Divergence relative to ancestor"</span>, <span class="at">color =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">Environment"</span>, <span class="at">fill =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">Environment"</span>) <span class="sc">+</span></span>
<span id="cb13-15"><a href="#cb13-15"></a>  <span class="co">#ggplot2::scale_y_sqrt() +</span></span>
<span id="cb13-16"><a href="#cb13-16"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">12</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">3.5</span>, <span class="fl">12.5</span>)) <span class="sc">+</span></span>
<span id="cb13-17"><a href="#cb13-17"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>mycols) <span class="sc">+</span></span>
<span id="cb13-18"><a href="#cb13-18"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values=</span>mycols) <span class="sc">+</span></span>
<span id="cb13-19"><a href="#cb13-19"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb13-20"><a href="#cb13-20"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb13-21"><a href="#cb13-21"></a>    <span class="co">#panel.grid.major = element_blank(),</span></span>
<span id="cb13-22"><a href="#cb13-22"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-23"><a href="#cb13-23"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-24"><a href="#cb13-24"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb13-25"><a href="#cb13-25"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-26"><a href="#cb13-26"></a>  )</span>
<span id="cb13-27"><a href="#cb13-27"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pjsd <span class="ot">&lt;-</span> <span class="fu">plot_jsd</span>(intrarepvar, jsddf_long_md_filt)</span>
<span id="cb14-2"><a href="#cb14-2"></a></span>
<span id="cb14-3"><a href="#cb14-3"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb14-4"><a href="#cb14-4"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_jsd_to_anc.svg"</span>),</span>
<span id="cb14-5"><a href="#cb14-5"></a>  pjsd,</span>
<span id="cb14-6"><a href="#cb14-6"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb14-7"><a href="#cb14-7"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb14-8"><a href="#cb14-8"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb14-9"><a href="#cb14-9"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb14-10"><a href="#cb14-10"></a>)</span>
<span id="cb14-11"><a href="#cb14-11"></a></span>
<span id="cb14-12"><a href="#cb14-12"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb14-13"><a href="#cb14-13"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_jsd_to_anc.png"</span>),</span>
<span id="cb14-14"><a href="#cb14-14"></a>  pjsd,</span>
<span id="cb14-15"><a href="#cb14-15"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb14-16"><a href="#cb14-16"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb14-17"><a href="#cb14-17"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb14-18"><a href="#cb14-18"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb14-19"><a href="#cb14-19"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_composition_analysis_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Mean <span class="math inline">\(\pm\)</span> 95% CI of the pairwise Jensen-Shannon Divergence between the ancestral/clonal community composition (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> C) and the community composition in YSK Home environments (Home, grid columns) for each measurement environment (Meas, grid rows) in the transplantation experiment (see also <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> D). Mean divergences are nested within Growth Cycles (horizontal axis) where one growth cycle is 7 days. Dashed lines indicate mean and ribbons indicate 95% CI for the mean within-condition Jensen-Shannon Divergence and is meant as an indicator of replicate variability within each Home/Measurement treatment combination.
</figcaption>
</figure>
</div>
</section>
<section id="jsd-to-ancestor---idiosyncratic-samples-filtered" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="jsd-to-ancestor---idiosyncratic-samples-filtered"><span class="header-section-number">6.2</span> JSD to ancestor - idiosyncratic samples filtered</h2>
<p>In <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> is it clear that some of the replicates in samples from YSK have a very different composition from the others. This is likely due to rapid resistance evolution of HAMBI 1972 in those particular replicates from YSK (will need to verify with the metagenomes). HAMBI_1972 is normally very sensitive to STR. The idea in this section is to repeat the JSD to ancestor analysis but exclude these idiosyncratic replicates - A and F from YSK BS, C and E from YSK B. We want to make sure that the trends in JSD relative to the ancestral community is not being driven primarily by these outlier replicates.</p>
<p>Filter JSD pairs to only those that are relative to ancestral community in each measurement environment. Also exclude the replicates mentioned above. Finally calculate mean and boostrapped 95CI using HMisc.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>jsddf_long_md_filt2 <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: BS"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"A|F"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"A|F"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: B"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"C|E"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"C|E"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-7"><a href="#cb15-7"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Calculating the within home/measurement divergence as some kind of variability baseline but with the outlier replicates excluded.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>intrarepvar2 <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: BS"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"A|F"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"A|F"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: B"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"C|E"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"C|E"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.y <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-7"><a href="#cb16-7"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Make plot of JSD to ancestral communities in each condition and save in raster/vector format</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>pjsd2 <span class="ot">&lt;-</span> <span class="fu">plot_jsd</span>(intrarepvar2, jsddf_long_md_filt2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_composition_analysis_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Same as in <a href="#fig-03" class="quarto-xref">Figure&nbsp;3</a> but with excluding the idiosyncratic replicates A and F from YSK Strep and C and E from YSK Bact.
</figcaption>
</figure>
</div>
<p>So here it does appear that the trends still hold but now the JSD to ancestor always exceeds variation within each environment/evolutionary history.</p>
</section>
<section id="jsd-to-ysk-founder-community" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="jsd-to-ysk-founder-community"><span class="header-section-number">6.3</span> JSD to YSK founder community</h2>
<p>Here I just wanted to check how each community with an YSK evolutionary history compares to the original YSK source material</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(transfer.x <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> transfer.y <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-3"><a href="#cb18-3"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_remove</span>(measure_env_short.y, <span class="st">"Meas: "</span>) <span class="sc">==</span> <span class="fu">str_remove</span>(home_env_short.y, <span class="st">"Home: "</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="co">#dplyr::filter(home_env_short.x == home_env_short.y) %&gt;% </span></span>
<span id="cb18-6"><a href="#cb18-6"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(replicate.x <span class="sc">==</span> replicate.y) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span> replicate.x, <span class="at">y =</span> jsd)) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(transfer.y)), <span class="at">height =</span> <span class="fl">0.01</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb18-9"><a href="#cb18-9"></a>  <span class="fu">facet_grid</span>(measure_env_short.y <span class="sc">~</span> home_env_short.x) <span class="sc">+</span></span>
<span id="cb18-10"><a href="#cb18-10"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Replicate microcosm"</span>, <span class="at">y =</span> <span class="st">"Jensen-Shannon Divergence relative YSK source"</span>, <span class="at">color =</span> <span class="st">"Transfer"</span>) <span class="sc">+</span></span>
<span id="cb18-11"><a href="#cb18-11"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb18-13"><a href="#cb18-13"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-14"><a href="#cb18-14"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-15"><a href="#cb18-15"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb18-16"><a href="#cb18-16"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-17"><a href="#cb18-17"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-05" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_composition_analysis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: JSD relative to 58 week YSK source material (nested by replicate) for each treatment/YSK history and transfer in the main experiment. Excludes sames that were inoculated with the ancestral community.
</figcaption>
</figure>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb19" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb19-1"><a href="#cb19-1"></a><span class="co">---</span></span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="an">title:</span><span class="co"> "Analysis of community compositions"</span></span>
<span id="cb19-3"><a href="#cb19-3"></a><span class="an">subtitle:</span><span class="co"> "From the STR community composition workflow"</span></span>
<span id="cb19-4"><a href="#cb19-4"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb19-5"><a href="#cb19-5"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb19-6"><a href="#cb19-6"></a><span class="an">abstract:</span><span class="co"> "This notebook plots community composition of the experiment, conducts ordination of CLR transformed count data, and compares the response of different YSK evolutionary histories to the ancestral community in each experimental measurement environment using pairwise Jensen-Shannon divergences."</span></span>
<span id="cb19-7"><a href="#cb19-7"></a><span class="co">---</span></span>
<span id="cb19-8"><a href="#cb19-8"></a></span>
<span id="cb19-9"><a href="#cb19-9"></a><span class="fu"># Setup</span></span>
<span id="cb19-10"><a href="#cb19-10"></a></span>
<span id="cb19-11"><a href="#cb19-11"></a>Loads required libraries and sets global variables</span>
<span id="cb19-12"><a href="#cb19-12"></a></span>
<span id="cb19-15"><a href="#cb19-15"></a><span class="in">```{r}</span></span>
<span id="cb19-16"><a href="#cb19-16"></a><span class="co">#| output: false</span></span>
<span id="cb19-17"><a href="#cb19-17"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb19-18"><a href="#cb19-18"></a><span class="fu">library</span>(here)</span>
<span id="cb19-19"><a href="#cb19-19"></a><span class="fu">library</span>(fs)</span>
<span id="cb19-20"><a href="#cb19-20"></a><span class="fu">library</span>(scales)</span>
<span id="cb19-21"><a href="#cb19-21"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb19-22"><a href="#cb19-22"></a><span class="fu">library</span>(Rtsne)</span>
<span id="cb19-23"><a href="#cb19-23"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb19-24"><a href="#cb19-24"></a><span class="fu">library</span>(philentropy)</span>
<span id="cb19-25"><a href="#cb19-25"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb19-26"><a href="#cb19-26"></a><span class="in">```</span></span>
<span id="cb19-27"><a href="#cb19-27"></a></span>
<span id="cb19-28"><a href="#cb19-28"></a><span class="fu"># Read</span></span>
<span id="cb19-29"><a href="#cb19-29"></a></span>
<span id="cb19-30"><a href="#cb19-30"></a>Read species abundance data (16S v3 amplicon counts) and do some light formatting of metadata </span>
<span id="cb19-31"><a href="#cb19-31"></a></span>
<span id="cb19-34"><a href="#cb19-34"></a><span class="in">```{r}</span></span>
<span id="cb19-35"><a href="#cb19-35"></a><span class="co">#| output: false</span></span>
<span id="cb19-36"><a href="#cb19-36"></a><span class="co">#| warning: false</span></span>
<span id="cb19-37"><a href="#cb19-37"></a>sptable <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_sp, <span class="st">"species_counts_md.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-38"><a href="#cb19-38"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> day<span class="sc">/</span><span class="dv">7</span>)</span>
<span id="cb19-39"><a href="#cb19-39"></a></span>
<span id="cb19-40"><a href="#cb19-40"></a>counts_f <span class="ot">&lt;-</span> sptable <span class="sc">%&gt;%</span> </span>
<span id="cb19-41"><a href="#cb19-41"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb19-42"><a href="#cb19-42"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">f=</span>count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-43"><a href="#cb19-43"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-44"><a href="#cb19-44"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">measure_env_short =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(measure_env <span class="sc">==</span> <span class="st">"bact"</span> <span class="sc">~</span> <span class="st">"Meas: B"</span>,</span>
<span id="cb19-45"><a href="#cb19-45"></a>                                       measure_env <span class="sc">==</span> <span class="st">"bact_strep"</span> <span class="sc">~</span> <span class="st">"Meas: BS"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-46"><a href="#cb19-46"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb19-47"><a href="#cb19-47"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"Home: Anc"</span>,</span>
<span id="cb19-48"><a href="#cb19-48"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"bact"</span> <span class="sc">~</span> <span class="st">"Home: B"</span>,</span>
<span id="cb19-49"><a href="#cb19-49"></a>                                    evolution_env <span class="sc">==</span> <span class="st">"bact_strep"</span> <span class="sc">~</span> <span class="st">"Home: BS"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-50"><a href="#cb19-50"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-51"><a href="#cb19-51"></a>    <span class="at">measure_env_short =</span> <span class="fu">factor</span>(measure_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Meas: B"</span>, <span class="st">"Meas: BS"</span>)),</span>
<span id="cb19-52"><a href="#cb19-52"></a>    <span class="at">home_env_short =</span> <span class="fu">factor</span>(home_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Home: Anc"</span>, <span class="st">"Home: B"</span>, <span class="st">"Home: BS"</span>)),</span>
<span id="cb19-53"><a href="#cb19-53"></a>         <span class="at">day =</span> <span class="fu">factor</span>(day),</span>
<span id="cb19-54"><a href="#cb19-54"></a>         <span class="at">replicate =</span> <span class="fu">factor</span>(replicate),</span>
<span id="cb19-55"><a href="#cb19-55"></a>         <span class="at">strainID =</span> <span class="fu">factor</span>(strainID, <span class="at">levels =</span> <span class="fu">names</span>(hambi_colors)))</span>
<span id="cb19-56"><a href="#cb19-56"></a><span class="in">```</span></span>
<span id="cb19-57"><a href="#cb19-57"></a></span>
<span id="cb19-58"><a href="#cb19-58"></a><span class="fu"># Formatting</span></span>
<span id="cb19-59"><a href="#cb19-59"></a></span>
<span id="cb19-60"><a href="#cb19-60"></a>Some light formatting to subset data into distint tibbles for later plotting</span>
<span id="cb19-61"><a href="#cb19-61"></a></span>
<span id="cb19-64"><a href="#cb19-64"></a><span class="in">```{r}</span></span>
<span id="cb19-65"><a href="#cb19-65"></a><span class="co">#| output: false</span></span>
<span id="cb19-66"><a href="#cb19-66"></a><span class="co">#| warning: false</span></span>
<span id="cb19-67"><a href="#cb19-67"></a><span class="co"># these are communities of a (supposedly) known composition. Can be used with metacal</span></span>
<span id="cb19-68"><a href="#cb19-68"></a>pos_ctrl_samples <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb19-69"><a href="#cb19-69"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(sample, <span class="st">"pos_ctrl"</span>))</span>
<span id="cb19-70"><a href="#cb19-70"></a></span>
<span id="cb19-71"><a href="#cb19-71"></a><span class="co"># these are samples taken directly from YSK and represent the composition of the communities used to start the experiment</span></span>
<span id="cb19-72"><a href="#cb19-72"></a>t0_samples <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb19-73"><a href="#cb19-73"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(sample, <span class="st">"pos_ctrl"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-74"><a href="#cb19-74"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(day <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb19-75"><a href="#cb19-75"></a></span>
<span id="cb19-76"><a href="#cb19-76"></a><span class="co"># only samples from the experiment</span></span>
<span id="cb19-77"><a href="#cb19-77"></a>counts_f_experiment <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">anti_join</span>(counts_f, pos_ctrl_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb19-78"><a href="#cb19-78"></a>  dplyr<span class="sc">::</span><span class="fu">anti_join</span>(., t0_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb19-79"><a href="#cb19-79"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-80"><a href="#cb19-80"></a>    <span class="at">measure_env_short =</span> <span class="fu">factor</span>(measure_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Meas: B"</span>, <span class="st">"Meas: BS"</span>)),</span>
<span id="cb19-81"><a href="#cb19-81"></a>    <span class="at">home_env_short =</span> <span class="fu">factor</span>(home_env_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Home: Anc"</span>, <span class="st">"Home: B"</span>, <span class="st">"Home: BS"</span>)),</span>
<span id="cb19-82"><a href="#cb19-82"></a>         <span class="at">day =</span> <span class="fu">factor</span>(day),</span>
<span id="cb19-83"><a href="#cb19-83"></a>         <span class="at">replicate =</span> <span class="fu">factor</span>(replicate),</span>
<span id="cb19-84"><a href="#cb19-84"></a>         <span class="at">strainID =</span> <span class="fu">factor</span>(strainID, <span class="at">levels =</span> <span class="fu">names</span>(hambi_colors)))</span>
<span id="cb19-85"><a href="#cb19-85"></a><span class="in">```</span></span>
<span id="cb19-86"><a href="#cb19-86"></a></span>
<span id="cb19-87"><a href="#cb19-87"></a><span class="fu"># Visualize community composition</span></span>
<span id="cb19-88"><a href="#cb19-88"></a></span>
<span id="cb19-89"><a href="#cb19-89"></a>Using ggplot2 to plot bar plots of species composition from the experiment and the intial material used to seed the experiment</span>
<span id="cb19-90"><a href="#cb19-90"></a></span>
<span id="cb19-93"><a href="#cb19-93"></a><span class="in">```{r}</span></span>
<span id="cb19-94"><a href="#cb19-94"></a><span class="co"># custom barplot theme</span></span>
<span id="cb19-95"><a href="#cb19-95"></a>mybartheme <span class="ot">&lt;-</span> <span class="cf">function</span>(...){</span>
<span id="cb19-96"><a href="#cb19-96"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-97"><a href="#cb19-97"></a>    <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>,<span class="st">"line"</span>),</span>
<span id="cb19-98"><a href="#cb19-98"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb19-99"><a href="#cb19-99"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-100"><a href="#cb19-100"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-101"><a href="#cb19-101"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-102"><a href="#cb19-102"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-103"><a href="#cb19-103"></a>    <span class="co">#axis.text.x = element_blank(),</span></span>
<span id="cb19-104"><a href="#cb19-104"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb19-105"><a href="#cb19-105"></a>    <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb19-106"><a href="#cb19-106"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-107"><a href="#cb19-107"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-108"><a href="#cb19-108"></a>    <span class="at">legend.key =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-109"><a href="#cb19-109"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb19-110"><a href="#cb19-110"></a>    ...)</span>
<span id="cb19-111"><a href="#cb19-111"></a>}</span>
<span id="cb19-112"><a href="#cb19-112"></a></span>
<span id="cb19-113"><a href="#cb19-113"></a><span class="co"># Base function for making the subplots</span></span>
<span id="cb19-114"><a href="#cb19-114"></a>basebar <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">x =</span> x, <span class="at">xlab =</span> xlab, <span class="at">xgrid =</span> <span class="cn">NULL</span>){</span>
<span id="cb19-115"><a href="#cb19-115"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(.data) <span class="sc">+</span></span>
<span id="cb19-116"><a href="#cb19-116"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> f, <span class="at">x =</span> {{ x }}, <span class="at">fill =</span> strainID), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">linewidth=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb19-117"><a href="#cb19-117"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> xlab, <span class="at">y=</span><span class="st">"% abundance"</span>, <span class="at">fill=</span><span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb19-118"><a href="#cb19-118"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb19-119"><a href="#cb19-119"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb19-120"><a href="#cb19-120"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> ggh4x<span class="sc">::</span><span class="fu">guide_axis_nested</span>()) <span class="sc">+</span></span>
<span id="cb19-121"><a href="#cb19-121"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb19-122"><a href="#cb19-122"></a>  <span class="fu">mybartheme</span>()</span>
<span id="cb19-123"><a href="#cb19-123"></a>}</span>
<span id="cb19-124"><a href="#cb19-124"></a></span>
<span id="cb19-125"><a href="#cb19-125"></a>A <span class="ot">&lt;-</span> t0_samples <span class="sc">%&gt;%</span> </span>
<span id="cb19-126"><a href="#cb19-126"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-127"><a href="#cb19-127"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> replicate, <span class="at">xlab =</span> <span class="st">"Replicate"</span>) <span class="sc">+</span></span>
<span id="cb19-128"><a href="#cb19-128"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>( <span class="sc">~</span> home_env_short)</span>
<span id="cb19-129"><a href="#cb19-129"></a></span>
<span id="cb19-130"><a href="#cb19-130"></a>B <span class="ot">&lt;-</span> t0_samples <span class="sc">%&gt;%</span> </span>
<span id="cb19-131"><a href="#cb19-131"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">!=</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-132"><a href="#cb19-132"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> replicate, <span class="at">xlab =</span> <span class="st">"Replicate"</span>) <span class="sc">+</span></span>
<span id="cb19-133"><a href="#cb19-133"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>( <span class="sc">~</span> home_env_short)</span>
<span id="cb19-134"><a href="#cb19-134"></a></span>
<span id="cb19-135"><a href="#cb19-135"></a>C <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb19-136"><a href="#cb19-136"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-137"><a href="#cb19-137"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> <span class="fu">interaction</span>(replicate, transfer), <span class="at">xlab =</span> <span class="st">"Replicate | Growth Cycle"</span>) <span class="sc">+</span></span>
<span id="cb19-138"><a href="#cb19-138"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short <span class="sc">~</span> home_env_short)</span>
<span id="cb19-139"><a href="#cb19-139"></a></span>
<span id="cb19-140"><a href="#cb19-140"></a>D <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb19-141"><a href="#cb19-141"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evolution_env <span class="sc">!=</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-142"><a href="#cb19-142"></a>  <span class="fu">basebar</span>(<span class="at">x =</span> <span class="fu">interaction</span>(replicate, transfer), <span class="at">xlab =</span> <span class="st">"Replicate | Growth Cycle"</span>) <span class="sc">+</span></span>
<span id="cb19-143"><a href="#cb19-143"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short <span class="sc">~</span> home_env_short)</span>
<span id="cb19-144"><a href="#cb19-144"></a><span class="in">```</span></span>
<span id="cb19-145"><a href="#cb19-145"></a></span>
<span id="cb19-146"><a href="#cb19-146"></a>Makes a final patchwork object and saves the results as raster and vector</span>
<span id="cb19-147"><a href="#cb19-147"></a></span>
<span id="cb19-150"><a href="#cb19-150"></a><span class="in">```{r}</span></span>
<span id="cb19-151"><a href="#cb19-151"></a><span class="co">#| warning: false</span></span>
<span id="cb19-152"><a href="#cb19-152"></a>layout <span class="ot">&lt;-</span> <span class="st">"</span></span>
<span id="cb19-153"><a href="#cb19-153"></a><span class="st">ABB</span></span>
<span id="cb19-154"><a href="#cb19-154"></a><span class="st">CDD</span></span>
<span id="cb19-155"><a href="#cb19-155"></a><span class="st">"</span></span>
<span id="cb19-156"><a href="#cb19-156"></a></span>
<span id="cb19-157"><a href="#cb19-157"></a>pbar <span class="ot">&lt;-</span> A <span class="sc">+</span> B <span class="sc">+</span> C <span class="sc">+</span> D  <span class="sc">+</span></span>
<span id="cb19-158"><a href="#cb19-158"></a>  patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.4</span>), </span>
<span id="cb19-159"><a href="#cb19-159"></a>              <span class="at">design =</span> layout) <span class="sc">+</span></span>
<span id="cb19-160"><a href="#cb19-160"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span></span>
<span id="cb19-161"><a href="#cb19-161"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">'bottom'</span>)</span>
<span id="cb19-162"><a href="#cb19-162"></a></span>
<span id="cb19-163"><a href="#cb19-163"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-164"><a href="#cb19-164"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_bar_plot.svg"</span>),</span>
<span id="cb19-165"><a href="#cb19-165"></a>  pbar,</span>
<span id="cb19-166"><a href="#cb19-166"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-167"><a href="#cb19-167"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb19-168"><a href="#cb19-168"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-169"><a href="#cb19-169"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb19-170"><a href="#cb19-170"></a>)</span>
<span id="cb19-171"><a href="#cb19-171"></a></span>
<span id="cb19-172"><a href="#cb19-172"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-173"><a href="#cb19-173"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_bar_plot.png"</span>),</span>
<span id="cb19-174"><a href="#cb19-174"></a>  pbar,</span>
<span id="cb19-175"><a href="#cb19-175"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-176"><a href="#cb19-176"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb19-177"><a href="#cb19-177"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-178"><a href="#cb19-178"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb19-179"><a href="#cb19-179"></a>)</span>
<span id="cb19-180"><a href="#cb19-180"></a><span class="in">```</span></span>
<span id="cb19-181"><a href="#cb19-181"></a></span>
<span id="cb19-182"><a href="#cb19-182"></a>::: {#fig-01}</span>
<span id="cb19-185"><a href="#cb19-185"></a><span class="in">```{r}</span></span>
<span id="cb19-186"><a href="#cb19-186"></a><span class="co">#| fig.width: 7</span></span>
<span id="cb19-187"><a href="#cb19-187"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb19-188"><a href="#cb19-188"></a><span class="co">#| echo: false</span></span>
<span id="cb19-189"><a href="#cb19-189"></a><span class="co">#| warning: false</span></span>
<span id="cb19-190"><a href="#cb19-190"></a>pbar</span>
<span id="cb19-191"><a href="#cb19-191"></a><span class="in">```</span></span>
<span id="cb19-192"><a href="#cb19-192"></a></span>
<span id="cb19-193"><a href="#cb19-193"></a>Relative abundance (Y axis, percent) of species from replicate microcosms (X axis, A-F). **A)** Community composition of the ancestral/clonal replicates inocula used to start experiments in C. **B)** Composition of the long-term community evolution experiment (YSK, week X) communities from each selection condition. These communities were used to inoculate the transplantation experiment in part D. **C)** Composition of the ancestral/clonal replicate communities after 4, 8, and 12 transfers in each measurement environment (grid rows). **D)** Composition of the YSK communities (grid columns, "home" environment) subjected to all possible measurement conditions (grid rows). B = bacteria grown without amendment, BS = streptomycin treatment, BP = *Tetrahymena thermophila* treatment, BPS = combined *Tetrahymena thermophila* and streptomycin treatment.</span>
<span id="cb19-194"><a href="#cb19-194"></a>:::</span>
<span id="cb19-195"><a href="#cb19-195"></a></span>
<span id="cb19-196"><a href="#cb19-196"></a><span class="fu"># tSNE dimension reduction of community composition</span></span>
<span id="cb19-197"><a href="#cb19-197"></a></span>
<span id="cb19-198"><a href="#cb19-198"></a>Some preliminary inspections of the number of samples with zeros. This is needed for zcompositions</span>
<span id="cb19-199"><a href="#cb19-199"></a></span>
<span id="cb19-202"><a href="#cb19-202"></a><span class="in">```{r}</span></span>
<span id="cb19-203"><a href="#cb19-203"></a>counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb19-204"><a href="#cb19-204"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb19-205"><a href="#cb19-205"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">n_samples =</span> <span class="fu">n</span>(),</span>
<span id="cb19-206"><a href="#cb19-206"></a>            <span class="at">n_gt0 =</span> <span class="fu">sum</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb19-207"><a href="#cb19-207"></a>            <span class="at">p_gt0 =</span> n_gt0 <span class="sc">/</span> n_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb19-208"><a href="#cb19-208"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-209"><a href="#cb19-209"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(n_gt0)</span>
<span id="cb19-210"><a href="#cb19-210"></a><span class="in">```</span></span>
<span id="cb19-211"><a href="#cb19-211"></a></span>
<span id="cb19-212"><a href="#cb19-212"></a>Here we will use the <span class="co">[</span><span class="ot">centered log-ratio transformation</span><span class="co">](https://en.wikipedia.org/wiki/Compositional_data#Center_log_ratio_transform)</span> for the species abundances. The centered log-ratio can be interpreted as the log-fold change of species i relative to the average microbe in a sample. The formula for the transformation is:</span>
<span id="cb19-213"><a href="#cb19-213"></a></span>
<span id="cb19-214"><a href="#cb19-214"></a>$$</span>
<span id="cb19-215"><a href="#cb19-215"></a>\text{clr}(\mathbf x)= \left(log</span>
<span id="cb19-216"><a href="#cb19-216"></a>\frac{x_i}{g(\mathbf x)} \right)_{i=1,...,D} \qquad \text{with} \quad</span>
<span id="cb19-217"><a href="#cb19-217"></a>g(\mathbf x) = \left(\prod_{i=1}^Dx_i\right)^{1/D} =</span>
<span id="cb19-218"><a href="#cb19-218"></a>\exp\left(\frac{1}{D}\sum_{i=1}^D \log x_i\right)\text{,}</span>
<span id="cb19-219"><a href="#cb19-219"></a>$$ {#eq-clr}</span>
<span id="cb19-220"><a href="#cb19-220"></a></span>
<span id="cb19-221"><a href="#cb19-221"></a>We will use the implementation of centered log-ratio transform in the <span class="in">`compositions`</span> package</span>
<span id="cb19-222"><a href="#cb19-222"></a></span>
<span id="cb19-225"><a href="#cb19-225"></a><span class="in">```{r}</span></span>
<span id="cb19-226"><a href="#cb19-226"></a><span class="fu">set.seed</span>(<span class="dv">234781</span>)</span>
<span id="cb19-227"><a href="#cb19-227"></a></span>
<span id="cb19-228"><a href="#cb19-228"></a><span class="co"># exclude these species because they have too many zeros</span></span>
<span id="cb19-229"><a href="#cb19-229"></a>lowstrainsv <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-230"><a href="#cb19-230"></a>  <span class="st">"HAMBI_0097"</span>,</span>
<span id="cb19-231"><a href="#cb19-231"></a>  <span class="st">"HAMBI_2792"</span></span>
<span id="cb19-232"><a href="#cb19-232"></a>)</span>
<span id="cb19-233"><a href="#cb19-233"></a></span>
<span id="cb19-234"><a href="#cb19-234"></a><span class="co"># make a species count matrix</span></span>
<span id="cb19-235"><a href="#cb19-235"></a>mymat <span class="ot">&lt;-</span> counts_f_experiment <span class="sc">%&gt;%</span> </span>
<span id="cb19-236"><a href="#cb19-236"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(strainID <span class="sc">%in%</span> lowstrainsv)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-237"><a href="#cb19-237"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count) <span class="sc">%&gt;%</span> </span>
<span id="cb19-238"><a href="#cb19-238"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count =</span> count <span class="sc">+</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-239"><a href="#cb19-239"></a>  <span class="co"># important to arrange by sample as this makes some later joins easier</span></span>
<span id="cb19-240"><a href="#cb19-240"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb19-241"><a href="#cb19-241"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"strainID"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-242"><a href="#cb19-242"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-243"><a href="#cb19-243"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb19-244"><a href="#cb19-244"></a></span>
<span id="cb19-245"><a href="#cb19-245"></a><span class="co"># calculate clr</span></span>
<span id="cb19-246"><a href="#cb19-246"></a>balclr <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(mymat)</span>
<span id="cb19-247"><a href="#cb19-247"></a></span>
<span id="cb19-248"><a href="#cb19-248"></a><span class="co"># tsne</span></span>
<span id="cb19-249"><a href="#cb19-249"></a>mymat_tsne <span class="ot">&lt;-</span> Rtsne<span class="sc">::</span><span class="fu">Rtsne</span>(<span class="fu">as.matrix</span>(balclr), <span class="at">perplexity =</span> <span class="dv">30</span>, <span class="at">dims =</span> <span class="dv">2</span>)</span>
<span id="cb19-250"><a href="#cb19-250"></a><span class="in">```</span></span>
<span id="cb19-251"><a href="#cb19-251"></a></span>
<span id="cb19-252"><a href="#cb19-252"></a>Define a plotting function to plot the tSNE output</span>
<span id="cb19-253"><a href="#cb19-253"></a></span>
<span id="cb19-256"><a href="#cb19-256"></a><span class="in">```{r}</span></span>
<span id="cb19-257"><a href="#cb19-257"></a>ptsne <span class="ot">&lt;-</span> <span class="cf">function</span>(tsne_obj, colorvar, shapevar, colors){</span>
<span id="cb19-258"><a href="#cb19-258"></a>  <span class="fu">data.frame</span>(tsne_obj<span class="sc">$</span>Y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-259"><a href="#cb19-259"></a>  <span class="fu">cbind</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(counts_f_experiment, sample, replicate, transfer, measure_env_short, home_env_short))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-260"><a href="#cb19-260"></a>  <span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="fu">factor</span>(transfer)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-261"><a href="#cb19-261"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb19-262"><a href="#cb19-262"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span> X2, <span class="at">y =</span> X1, <span class="at">color =</span> {{ colorvar }}, <span class="at">shape =</span> {{ shapevar }}), <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-263"><a href="#cb19-263"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb19-264"><a href="#cb19-264"></a>  ggplot2<span class="sc">::</span><span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb19-265"><a href="#cb19-265"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"tSNE dimension 1"</span>, </span>
<span id="cb19-266"><a href="#cb19-266"></a>       <span class="at">y =</span> <span class="st">"tSNE dimension 2"</span>, </span>
<span id="cb19-267"><a href="#cb19-267"></a>       <span class="at">shape =</span> <span class="st">"Transfer"</span>) <span class="sc">+</span></span>
<span id="cb19-268"><a href="#cb19-268"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> colors) <span class="sc">+</span></span>
<span id="cb19-269"><a href="#cb19-269"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb19-270"><a href="#cb19-270"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-271"><a href="#cb19-271"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-272"><a href="#cb19-272"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-273"><a href="#cb19-273"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-274"><a href="#cb19-274"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb19-275"><a href="#cb19-275"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-276"><a href="#cb19-276"></a>  ) </span>
<span id="cb19-277"><a href="#cb19-277"></a>}</span>
<span id="cb19-278"><a href="#cb19-278"></a><span class="in">```</span></span>
<span id="cb19-279"><a href="#cb19-279"></a></span>
<span id="cb19-280"><a href="#cb19-280"></a>Call the plotting function, make the patchwork layout, and save in raster and vector format</span>
<span id="cb19-281"><a href="#cb19-281"></a></span>
<span id="cb19-284"><a href="#cb19-284"></a><span class="in">```{r}</span></span>
<span id="cb19-285"><a href="#cb19-285"></a>pm <span class="ot">&lt;-</span> <span class="fu">ptsne</span>(mymat_tsne, measure_env_short, <span class="at">shapevar =</span> transfer, <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>)) <span class="sc">+</span></span>
<span id="cb19-286"><a href="#cb19-286"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Measure</span><span class="sc">\n</span><span class="st">environment"</span>)</span>
<span id="cb19-287"><a href="#cb19-287"></a>ph <span class="ot">&lt;-</span> <span class="fu">ptsne</span>(mymat_tsne, home_env_short, <span class="at">shapevar =</span> transfer, <span class="fu">c</span>(<span class="st">"#8da0cb"</span>, <span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>)) <span class="sc">+</span></span>
<span id="cb19-288"><a href="#cb19-288"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">environment"</span>)</span>
<span id="cb19-289"><a href="#cb19-289"></a></span>
<span id="cb19-290"><a href="#cb19-290"></a>pt <span class="ot">&lt;-</span> pm <span class="sc">+</span> ph <span class="sc">+</span></span>
<span id="cb19-291"><a href="#cb19-291"></a>  patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb19-292"><a href="#cb19-292"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>)</span>
<span id="cb19-293"><a href="#cb19-293"></a></span>
<span id="cb19-294"><a href="#cb19-294"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-295"><a href="#cb19-295"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_tsne.svg"</span>),</span>
<span id="cb19-296"><a href="#cb19-296"></a>  pt,</span>
<span id="cb19-297"><a href="#cb19-297"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-298"><a href="#cb19-298"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb19-299"><a href="#cb19-299"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-300"><a href="#cb19-300"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb19-301"><a href="#cb19-301"></a>)</span>
<span id="cb19-302"><a href="#cb19-302"></a></span>
<span id="cb19-303"><a href="#cb19-303"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-304"><a href="#cb19-304"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_composition_tsne.png"</span>),</span>
<span id="cb19-305"><a href="#cb19-305"></a>  pt,</span>
<span id="cb19-306"><a href="#cb19-306"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-307"><a href="#cb19-307"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb19-308"><a href="#cb19-308"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-309"><a href="#cb19-309"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb19-310"><a href="#cb19-310"></a>)</span>
<span id="cb19-311"><a href="#cb19-311"></a><span class="in">```</span></span>
<span id="cb19-312"><a href="#cb19-312"></a></span>
<span id="cb19-313"><a href="#cb19-313"></a>::: {#fig-02}</span>
<span id="cb19-316"><a href="#cb19-316"></a><span class="in">```{r}</span></span>
<span id="cb19-317"><a href="#cb19-317"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb19-318"><a href="#cb19-318"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb19-319"><a href="#cb19-319"></a><span class="co">#| echo: false</span></span>
<span id="cb19-320"><a href="#cb19-320"></a><span class="co">#| warning: false</span></span>
<span id="cb19-321"><a href="#cb19-321"></a>pt</span>
<span id="cb19-322"><a href="#cb19-322"></a><span class="in">```</span></span>
<span id="cb19-323"><a href="#cb19-323"></a></span>
<span id="cb19-324"><a href="#cb19-324"></a>Dimensional reduction of bacterial community composition using t-SNE. Each point represents a replicate microcosm sampled at serial transfers 4, 8, and 12 (grid columns). In **A)** point colors depict the experimental measurement condition from the transplantation experiment. In **B)** point colors depict the evolutionary history of the inoculating communities ("home environment").</span>
<span id="cb19-325"><a href="#cb19-325"></a>:::</span>
<span id="cb19-326"><a href="#cb19-326"></a></span>
<span id="cb19-327"><a href="#cb19-327"></a><span class="fu"># Distances between communities</span></span>
<span id="cb19-328"><a href="#cb19-328"></a></span>
<span id="cb19-329"><a href="#cb19-329"></a>The Jensen–Shannon divergence (JSD) is a symmetrized and smoothed version of the <span class="co">[</span><span class="ot">Kullback–Leibler divergence</span><span class="co">](https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence "Kullback–Leibler divergence")</span>. It is defined as:</span>
<span id="cb19-330"><a href="#cb19-330"></a></span>
<span id="cb19-331"><a href="#cb19-331"></a>$$</span>
<span id="cb19-332"><a href="#cb19-332"></a>D_{\text{JS}}(P \parallel Q) = \frac{1}{2} D_{\text{KL}}(P \parallel M) + \frac{1}{2} D_{\text{KL}}(Q \parallel M)</span>
<span id="cb19-333"><a href="#cb19-333"></a>$$ {#eq-JSD}</span>
<span id="cb19-334"><a href="#cb19-334"></a></span>
<span id="cb19-335"><a href="#cb19-335"></a>$$</span>
<span id="cb19-336"><a href="#cb19-336"></a>M = \frac{1}{2}(P + Q)</span>
<span id="cb19-337"><a href="#cb19-337"></a>$$ {#eq-avg_dist}</span>
<span id="cb19-338"><a href="#cb19-338"></a></span>
<span id="cb19-339"><a href="#cb19-339"></a>Here, ($D_{\text{KL}}$) denotes the Kullback-Leibler divergence which is defined as:</span>
<span id="cb19-340"><a href="#cb19-340"></a></span>
<span id="cb19-341"><a href="#cb19-341"></a>$$</span>
<span id="cb19-342"><a href="#cb19-342"></a>D_{KL}(P \parallel Q) = \sum_{x \in \mathcal{X}} P(x) \log \frac{P(x)}{Q(x)}</span>
<span id="cb19-343"><a href="#cb19-343"></a>$$ {#eq-KLD}</span>
<span id="cb19-344"><a href="#cb19-344"></a></span>
<span id="cb19-345"><a href="#cb19-345"></a><span class="fu">## JSD to ancestor no samples filtered</span></span>
<span id="cb19-346"><a href="#cb19-346"></a></span>
<span id="cb19-349"><a href="#cb19-349"></a><span class="in">```{r}</span></span>
<span id="cb19-350"><a href="#cb19-350"></a><span class="co">#| output: false</span></span>
<span id="cb19-351"><a href="#cb19-351"></a><span class="co">#| warning: false</span></span>
<span id="cb19-352"><a href="#cb19-352"></a><span class="co"># create a matrix of species frequencies</span></span>
<span id="cb19-353"><a href="#cb19-353"></a>prob_mat <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb19-354"><a href="#cb19-354"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f) <span class="sc">%&gt;%</span> </span>
<span id="cb19-355"><a href="#cb19-355"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> strainID, <span class="at">values_from =</span> f) <span class="sc">%&gt;%</span> </span>
<span id="cb19-356"><a href="#cb19-356"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-357"><a href="#cb19-357"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb19-358"><a href="#cb19-358"></a></span>
<span id="cb19-359"><a href="#cb19-359"></a><span class="co"># calculate JSD between all pairs using philentropy::JSD</span></span>
<span id="cb19-360"><a href="#cb19-360"></a>jsd_mat <span class="ot">&lt;-</span> philentropy<span class="sc">::</span><span class="fu">JSD</span>(prob_mat)</span>
<span id="cb19-361"><a href="#cb19-361"></a></span>
<span id="cb19-362"><a href="#cb19-362"></a><span class="co"># get nonredundant pairings</span></span>
<span id="cb19-363"><a href="#cb19-363"></a>ind <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">upper.tri</span>(jsd_mat, <span class="at">diag =</span> <span class="cn">TRUE</span>), <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-364"><a href="#cb19-364"></a>nn <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">rownames</span>(prob_mat), <span class="fu">rownames</span>(prob_mat))</span>
<span id="cb19-365"><a href="#cb19-365"></a>jsddf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample.x =</span> nn[[<span class="dv">1</span>]][ind[, <span class="dv">1</span>]],</span>
<span id="cb19-366"><a href="#cb19-366"></a>           <span class="at">sample.y =</span> nn[[<span class="dv">2</span>]][ind[, <span class="dv">2</span>]],</span>
<span id="cb19-367"><a href="#cb19-367"></a>           <span class="at">jsd =</span> jsd_mat[ind])</span>
<span id="cb19-368"><a href="#cb19-368"></a></span>
<span id="cb19-369"><a href="#cb19-369"></a><span class="co"># metadata for joining</span></span>
<span id="cb19-370"><a href="#cb19-370"></a>md <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(counts_f, sample, replicate, transfer, measure_env_short, home_env_short)</span>
<span id="cb19-371"><a href="#cb19-371"></a></span>
<span id="cb19-372"><a href="#cb19-372"></a><span class="co"># convert to long format</span></span>
<span id="cb19-373"><a href="#cb19-373"></a>jsddf_long_md <span class="ot">&lt;-</span> jsddf <span class="sc">%&gt;%</span> </span>
<span id="cb19-374"><a href="#cb19-374"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(md, <span class="at">by =</span> <span class="fu">join_by</span>(sample.x<span class="sc">==</span>sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-375"><a href="#cb19-375"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(md, <span class="at">by =</span> <span class="fu">join_by</span>(sample.y<span class="sc">==</span>sample))</span>
<span id="cb19-376"><a href="#cb19-376"></a><span class="in">```</span></span>
<span id="cb19-377"><a href="#cb19-377"></a></span>
<span id="cb19-378"><a href="#cb19-378"></a>Filter JSD pairs to only those that are relative to ancestral community in each measurement environment. Also calculate mean and boostrapped 95CI using HMisc</span>
<span id="cb19-379"><a href="#cb19-379"></a></span>
<span id="cb19-382"><a href="#cb19-382"></a><span class="in">```{r}</span></span>
<span id="cb19-383"><a href="#cb19-383"></a><span class="co">#| output: false</span></span>
<span id="cb19-384"><a href="#cb19-384"></a><span class="co">#| warning: false</span></span>
<span id="cb19-385"><a href="#cb19-385"></a>jsddf_long_md_filt <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb19-386"><a href="#cb19-386"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-387"><a href="#cb19-387"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-388"><a href="#cb19-388"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-389"><a href="#cb19-389"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span>
<span id="cb19-390"><a href="#cb19-390"></a><span class="in">```</span></span>
<span id="cb19-391"><a href="#cb19-391"></a></span>
<span id="cb19-392"><a href="#cb19-392"></a>Hmmm.... not sure what exactly this is telling, but I am calculating the within home/measurement divergence as some kind of variability baseline.</span>
<span id="cb19-393"><a href="#cb19-393"></a></span>
<span id="cb19-396"><a href="#cb19-396"></a><span class="in">```{r}</span></span>
<span id="cb19-397"><a href="#cb19-397"></a>intrarepvar <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb19-398"><a href="#cb19-398"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-399"><a href="#cb19-399"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-400"><a href="#cb19-400"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.y <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-401"><a href="#cb19-401"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span>
<span id="cb19-402"><a href="#cb19-402"></a><span class="in">```</span></span>
<span id="cb19-403"><a href="#cb19-403"></a></span>
<span id="cb19-404"><a href="#cb19-404"></a>Make plot of JSD to ancestral communities in each condition and save in raster/vector format</span>
<span id="cb19-405"><a href="#cb19-405"></a></span>
<span id="cb19-408"><a href="#cb19-408"></a><span class="in">```{r}</span></span>
<span id="cb19-409"><a href="#cb19-409"></a>plot_jsd <span class="ot">&lt;-</span> <span class="cf">function</span>(intra_df, df){</span>
<span id="cb19-410"><a href="#cb19-410"></a>  mycols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Home: B"</span> <span class="ot">=</span> <span class="st">"#66c2a5"</span>, <span class="st">"Home: BS"</span> <span class="ot">=</span> <span class="st">"#fc8d62"</span>)</span>
<span id="cb19-411"><a href="#cb19-411"></a>  ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb19-412"><a href="#cb19-412"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_ribbon</span>(<span class="at">data =</span> intra_df, </span>
<span id="cb19-413"><a href="#cb19-413"></a>              <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">ymin =</span> ymin, <span class="at">ymax=</span>ymax, <span class="at">fill =</span> home_env_short.y), <span class="at">alpha=</span><span class="fl">0.25</span>) <span class="sc">+</span> </span>
<span id="cb19-414"><a href="#cb19-414"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">data =</span> intra_df, </span>
<span id="cb19-415"><a href="#cb19-415"></a>            <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y=</span>y, <span class="at">color =</span> home_env_short.y), <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">lty=</span><span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb19-416"><a href="#cb19-416"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y =</span> y, <span class="at">group =</span> home_env_short.y, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span></span>
<span id="cb19-417"><a href="#cb19-417"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_linerange</span>(<span class="at">data =</span> df, </span>
<span id="cb19-418"><a href="#cb19-418"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">ymin =</span> ymin, <span class="at">ymax =</span> ymax, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span> </span>
<span id="cb19-419"><a href="#cb19-419"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">data=</span> df, </span>
<span id="cb19-420"><a href="#cb19-420"></a>            <span class="fu">aes</span>(<span class="at">x =</span> transfer.y, <span class="at">y =</span> y, <span class="at">group =</span> home_env_short.y, <span class="at">color =</span> home_env_short.y)) <span class="sc">+</span> </span>
<span id="cb19-421"><a href="#cb19-421"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(measure_env_short.y <span class="sc">~</span> home_env_short.y, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span> </span>
<span id="cb19-422"><a href="#cb19-422"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Growth Cycles"</span>, <span class="at">y =</span> <span class="st">"Jensen-Shannon Divergence relative to ancestor"</span>, <span class="at">color =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">Environment"</span>, <span class="at">fill =</span> <span class="st">"Home</span><span class="sc">\n</span><span class="st">Environment"</span>) <span class="sc">+</span></span>
<span id="cb19-423"><a href="#cb19-423"></a>  <span class="co">#ggplot2::scale_y_sqrt() +</span></span>
<span id="cb19-424"><a href="#cb19-424"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">8</span>, <span class="dv">12</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">3.5</span>, <span class="fl">12.5</span>)) <span class="sc">+</span></span>
<span id="cb19-425"><a href="#cb19-425"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values=</span>mycols) <span class="sc">+</span></span>
<span id="cb19-426"><a href="#cb19-426"></a>  ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(<span class="at">values=</span>mycols) <span class="sc">+</span></span>
<span id="cb19-427"><a href="#cb19-427"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb19-428"><a href="#cb19-428"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-429"><a href="#cb19-429"></a>    <span class="co">#panel.grid.major = element_blank(),</span></span>
<span id="cb19-430"><a href="#cb19-430"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-431"><a href="#cb19-431"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-432"><a href="#cb19-432"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb19-433"><a href="#cb19-433"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-434"><a href="#cb19-434"></a>  )</span>
<span id="cb19-435"><a href="#cb19-435"></a>}</span>
<span id="cb19-436"><a href="#cb19-436"></a><span class="in">```</span></span>
<span id="cb19-437"><a href="#cb19-437"></a></span>
<span id="cb19-440"><a href="#cb19-440"></a><span class="in">```{r}</span></span>
<span id="cb19-441"><a href="#cb19-441"></a>pjsd <span class="ot">&lt;-</span> <span class="fu">plot_jsd</span>(intrarepvar, jsddf_long_md_filt)</span>
<span id="cb19-442"><a href="#cb19-442"></a></span>
<span id="cb19-443"><a href="#cb19-443"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-444"><a href="#cb19-444"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_jsd_to_anc.svg"</span>),</span>
<span id="cb19-445"><a href="#cb19-445"></a>  pjsd,</span>
<span id="cb19-446"><a href="#cb19-446"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-447"><a href="#cb19-447"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb19-448"><a href="#cb19-448"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-449"><a href="#cb19-449"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb19-450"><a href="#cb19-450"></a>)</span>
<span id="cb19-451"><a href="#cb19-451"></a></span>
<span id="cb19-452"><a href="#cb19-452"></a>ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb19-453"><a href="#cb19-453"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"community_jsd_to_anc.png"</span>),</span>
<span id="cb19-454"><a href="#cb19-454"></a>  pjsd,</span>
<span id="cb19-455"><a href="#cb19-455"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb19-456"><a href="#cb19-456"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb19-457"><a href="#cb19-457"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb19-458"><a href="#cb19-458"></a>  <span class="at">device =</span> <span class="st">"png"</span></span>
<span id="cb19-459"><a href="#cb19-459"></a>)</span>
<span id="cb19-460"><a href="#cb19-460"></a><span class="in">```</span></span>
<span id="cb19-461"><a href="#cb19-461"></a></span>
<span id="cb19-462"><a href="#cb19-462"></a>::: {#fig-03}</span>
<span id="cb19-465"><a href="#cb19-465"></a><span class="in">```{r}</span></span>
<span id="cb19-466"><a href="#cb19-466"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb19-467"><a href="#cb19-467"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb19-468"><a href="#cb19-468"></a><span class="co">#| echo: false</span></span>
<span id="cb19-469"><a href="#cb19-469"></a><span class="co">#| warning: false</span></span>
<span id="cb19-470"><a href="#cb19-470"></a>pjsd</span>
<span id="cb19-471"><a href="#cb19-471"></a><span class="in">```</span></span>
<span id="cb19-472"><a href="#cb19-472"></a>Mean $\pm$ 95% CI of the pairwise Jensen-Shannon Divergence between the ancestral/clonal community composition (@fig-01 C) and the community composition in YSK Home environments (Home, grid columns) for each measurement environment (Meas, grid rows) in the transplantation experiment (see also @fig-01 D). Mean divergences are nested within Growth Cycles (horizontal axis) where one growth cycle is 7 days. Dashed lines indicate mean and ribbons indicate 95% CI for the mean within-condition Jensen-Shannon Divergence and is meant as an indicator of replicate variability within each Home/Measurement treatment combination.</span>
<span id="cb19-473"><a href="#cb19-473"></a>:::</span>
<span id="cb19-474"><a href="#cb19-474"></a></span>
<span id="cb19-475"><a href="#cb19-475"></a><span class="fu">## JSD to ancestor - idiosyncratic samples filtered </span></span>
<span id="cb19-476"><a href="#cb19-476"></a></span>
<span id="cb19-477"><a href="#cb19-477"></a>In @fig-01 is it clear that some of the replicates in samples from YSK have a very different composition from the others. This is likely due to rapid resistance evolution of HAMBI 1972 in those particular replicates from YSK (will need to verify with the metagenomes). HAMBI_1972 is normally very sensitive to STR. The idea in this section is to repeat the JSD to ancestor analysis but exclude these idiosyncratic replicates - A and F from YSK BS, C and E from YSK B. We want to make sure that the trends in JSD relative to the ancestral community is not being driven primarily by these outlier replicates.</span>
<span id="cb19-478"><a href="#cb19-478"></a></span>
<span id="cb19-479"><a href="#cb19-479"></a>Filter JSD pairs to only those that are relative to ancestral community in each measurement environment. Also exclude the replicates mentioned above. Finally calculate mean and boostrapped 95CI using HMisc.</span>
<span id="cb19-480"><a href="#cb19-480"></a></span>
<span id="cb19-483"><a href="#cb19-483"></a><span class="in">```{r}</span></span>
<span id="cb19-484"><a href="#cb19-484"></a><span class="co">#| output: false</span></span>
<span id="cb19-485"><a href="#cb19-485"></a><span class="co">#| warning: false</span></span>
<span id="cb19-486"><a href="#cb19-486"></a>jsddf_long_md_filt2 <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb19-487"><a href="#cb19-487"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-488"><a href="#cb19-488"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-489"><a href="#cb19-489"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: BS"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"A|F"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"A|F"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-490"><a href="#cb19-490"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: B"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"C|E"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"C|E"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-491"><a href="#cb19-491"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-492"><a href="#cb19-492"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span>
<span id="cb19-493"><a href="#cb19-493"></a>  </span>
<span id="cb19-494"><a href="#cb19-494"></a><span class="in">```</span></span>
<span id="cb19-495"><a href="#cb19-495"></a></span>
<span id="cb19-496"><a href="#cb19-496"></a>Calculating the within home/measurement divergence as some kind of variability baseline but with the outlier replicates excluded.</span>
<span id="cb19-497"><a href="#cb19-497"></a></span>
<span id="cb19-500"><a href="#cb19-500"></a><span class="in">```{r}</span></span>
<span id="cb19-501"><a href="#cb19-501"></a>intrarepvar2 <span class="ot">&lt;-</span> jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb19-502"><a href="#cb19-502"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(measure_env_short.x <span class="sc">==</span> measure_env_short.y <span class="sc">&amp;</span> home_env_short.x <span class="sc">==</span> home_env_short.y <span class="sc">&amp;</span> transfer.x <span class="sc">==</span> transfer.y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-503"><a href="#cb19-503"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: BS"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"A|F"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"A|F"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-504"><a href="#cb19-504"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>((measure_env_short.x <span class="sc">==</span> <span class="st">"Meas: BS"</span> <span class="sc">&amp;</span> home_env_short.y <span class="sc">==</span> <span class="st">"Home: B"</span>) <span class="sc">&amp;</span> <span class="fu">str_detect</span>(replicate.x, <span class="st">"C|E"</span>) <span class="sc">|</span> <span class="fu">str_detect</span>(replicate.y, <span class="st">"C|E"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-505"><a href="#cb19-505"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">mean_cl_boot</span>(jsd), <span class="at">.by=</span><span class="fu">c</span>(transfer.y, home_env_short.y, measure_env_short.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-506"><a href="#cb19-506"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.y <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-507"><a href="#cb19-507"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">home_env_short.y =</span> <span class="fu">factor</span>(home_env_short.y))</span>
<span id="cb19-508"><a href="#cb19-508"></a><span class="in">```</span></span>
<span id="cb19-509"><a href="#cb19-509"></a></span>
<span id="cb19-510"><a href="#cb19-510"></a>Make plot of JSD to ancestral communities in each condition and save in raster/vector format</span>
<span id="cb19-511"><a href="#cb19-511"></a></span>
<span id="cb19-514"><a href="#cb19-514"></a><span class="in">```{r}</span></span>
<span id="cb19-515"><a href="#cb19-515"></a>pjsd2 <span class="ot">&lt;-</span> <span class="fu">plot_jsd</span>(intrarepvar2, jsddf_long_md_filt2)</span>
<span id="cb19-516"><a href="#cb19-516"></a><span class="in">```</span></span>
<span id="cb19-517"><a href="#cb19-517"></a></span>
<span id="cb19-518"><a href="#cb19-518"></a>::: {#fig-04}</span>
<span id="cb19-521"><a href="#cb19-521"></a><span class="in">```{r}</span></span>
<span id="cb19-522"><a href="#cb19-522"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb19-523"><a href="#cb19-523"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb19-524"><a href="#cb19-524"></a><span class="co">#| echo: false</span></span>
<span id="cb19-525"><a href="#cb19-525"></a><span class="co">#| warning: false</span></span>
<span id="cb19-526"><a href="#cb19-526"></a>pjsd2</span>
<span id="cb19-527"><a href="#cb19-527"></a><span class="in">```</span></span>
<span id="cb19-528"><a href="#cb19-528"></a>Same as in @fig-03 but with excluding the idiosyncratic replicates A and F from YSK Strep and C and E from YSK Bact.</span>
<span id="cb19-529"><a href="#cb19-529"></a>:::</span>
<span id="cb19-530"><a href="#cb19-530"></a></span>
<span id="cb19-531"><a href="#cb19-531"></a>So here it does appear that the trends still hold but now the JSD to ancestor always exceeds variation within each environment/evolutionary history.</span>
<span id="cb19-532"><a href="#cb19-532"></a></span>
<span id="cb19-533"><a href="#cb19-533"></a><span class="fu">## JSD to YSK founder community</span></span>
<span id="cb19-534"><a href="#cb19-534"></a></span>
<span id="cb19-535"><a href="#cb19-535"></a>Here I just wanted to check how each community with an YSK evolutionary history compares to the original YSK source material</span>
<span id="cb19-536"><a href="#cb19-536"></a></span>
<span id="cb19-537"><a href="#cb19-537"></a>::: {#fig-05}</span>
<span id="cb19-540"><a href="#cb19-540"></a><span class="in">```{r}</span></span>
<span id="cb19-541"><a href="#cb19-541"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb19-542"><a href="#cb19-542"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb19-543"><a href="#cb19-543"></a><span class="co">#| warning: false</span></span>
<span id="cb19-544"><a href="#cb19-544"></a>jsddf_long_md <span class="sc">%&gt;%</span> </span>
<span id="cb19-545"><a href="#cb19-545"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(transfer.x <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> transfer.y <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-546"><a href="#cb19-546"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(home_env_short.x <span class="sc">!=</span> <span class="st">"Home: Anc"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-547"><a href="#cb19-547"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_remove</span>(measure_env_short.y, <span class="st">"Meas: "</span>) <span class="sc">==</span> <span class="fu">str_remove</span>(home_env_short.y, <span class="st">"Home: "</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-548"><a href="#cb19-548"></a>  <span class="co">#dplyr::filter(home_env_short.x == home_env_short.y) %&gt;% </span></span>
<span id="cb19-549"><a href="#cb19-549"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(replicate.x <span class="sc">==</span> replicate.y) <span class="sc">%&gt;%</span> </span>
<span id="cb19-550"><a href="#cb19-550"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span> replicate.x, <span class="at">y =</span> jsd)) <span class="sc">+</span></span>
<span id="cb19-551"><a href="#cb19-551"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> <span class="fu">factor</span>(transfer.y)), <span class="at">height =</span> <span class="fl">0.01</span>, <span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-552"><a href="#cb19-552"></a>  <span class="fu">facet_grid</span>(measure_env_short.y <span class="sc">~</span> home_env_short.x) <span class="sc">+</span></span>
<span id="cb19-553"><a href="#cb19-553"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Replicate microcosm"</span>, <span class="at">y =</span> <span class="st">"Jensen-Shannon Divergence relative YSK source"</span>, <span class="at">color =</span> <span class="st">"Transfer"</span>) <span class="sc">+</span></span>
<span id="cb19-554"><a href="#cb19-554"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb19-555"><a href="#cb19-555"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-556"><a href="#cb19-556"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-557"><a href="#cb19-557"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-558"><a href="#cb19-558"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb19-559"><a href="#cb19-559"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-560"><a href="#cb19-560"></a>  )</span>
<span id="cb19-561"><a href="#cb19-561"></a></span>
<span id="cb19-562"><a href="#cb19-562"></a><span class="in">```</span></span>
<span id="cb19-563"><a href="#cb19-563"></a>JSD relative to 58 week YSK source material (nested by replicate) for each treatment/YSK history and transfer in the main experiment. Excludes sames that were inoculated with the ancestral community.</span>
<span id="cb19-564"><a href="#cb19-564"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>